<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2022-02-21" />

<title>Captain’s Log</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/theandyb/phasing">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Captain’s Log</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2022-02-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>phasing/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220221code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220221)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220221code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220221)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingtree69bf5631de92053ec758e10babe29a9578246314targetblank69bf563a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/phasing/tree/69bf5631de92053ec758e10babe29a9578246314" target="_blank">69bf563</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingtree69bf5631de92053ec758e10babe29a9578246314targetblank69bf563a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/phasing/tree/69bf5631de92053ec758e10babe29a9578246314" target="_blank">69bf563</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/1kgp/
    Ignored:    data/consensus/
    Ignored:    data/gnomAD/
    Ignored:    data/mask/
    Ignored:    data/ovary/
    Ignored:    data/recomb/
    Ignored:    data/ref/
    Ignored:    data/shapeit/
    Ignored:    output/2023_switch_errors/
    Ignored:    output/X_hets/
    Ignored:    output/admix_switch_errors/
    Ignored:    output/afr_eur_ref/
    Ignored:    output/background_rates/
    Ignored:    output/figures/
    Ignored:    output/filter_switch_errors/
    Ignored:    output/final_switch_errors/
    Ignored:    output/gnomad/
    Ignored:    output/male_only_reference/
    Ignored:    output/nygc/
    Ignored:    output/singleton_switch_errors/
    Ignored:    plink.log
    Ignored:    sandbox/

Untracked files:
    Untracked:  analysis/genomic_features.Rmd

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/all_maf_plots.Rmd
    Modified:   code/batch_annotate_het.sh
    Modified:   code/batch_annotate_switch.sh
    Modified:   code/batch_diploid_v3.sh
    Modified:   code/batch_vote_phase.sh
    Modified:   code/common_functions.R
    Deleted:    code/core

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/captains_log.Rmd</code>) and HTML
(<code>docs/captains_log.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing/blob/69bf5631de92053ec758e10babe29a9578246314/analysis/captains_log.Rmd" target="_blank">69bf563</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-02-13
</td>
<td>
wflow_publish("analysis/captains_log.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing/blob/09ce177ab51a198102f3eb5e930e5680e8d28a52/analysis/captains_log.Rmd" target="_blank">09ce177</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-09-14
</td>
<td>
repo as of 14-sept-23
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing/blob/e239a478a3e38767c6be48c225be9977e624f665/analysis/captains_log.Rmd" target="_blank">e239a47</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-02-15
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing/e239a478a3e38767c6be48c225be9977e624f665/docs/captains_log.html" target="_blank">e239a47</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-02-15
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing/0b8c9e718ff8981e197e7304976d650761a863bb/docs/captains_log.html" target="_blank">0b8c9e7</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing/blob/72ed6c4029027940fcb8b5866fe6025cf4b63a8b/analysis/captains_log.Rmd" target="_blank">72ed6c4</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-02-28
</td>
<td>
wflow_publish(c("analysis/captains_log.Rmd", "analysis/build_a_vcf.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here in this document I will be documenting every action that I
perform for this ill-defined “phasing project.” Onwards!</p>
</div>
<div id="feb-2022" class="section level2">
<h2>21 Feb 2022</h2>
<div id="installation-of-shapeit4" class="section level3">
<h3>Installation of SHAPEIT4</h3>
<p>Following the instructions that I found at <a
href="https://odelaneau.github.io/shapeit4/#installation">this site</a>,
I</p>
<ol style="list-style-type: decimal">
<li>Downloaded the source code:
<code>wget https://github.com/odelaneau/shapeit4/archive/refs/tags/v4.2.2.tar.gz</code></li>
<li>Extracted the source code: <code>tar zxvf v4.2.2.tar.gz</code></li>
<li>Per instructions edited the paths in the <code>makefile</code>
<ul>
<li>Should I have installed my own version of HTSlib? -&gt; edit: going
to do this, under $HOME/Tools</li>
<li>BOOST paths did not need to be changed</li>
</ul></li>
<li>Compile using <code>make</code></li>
<li>Add path to bin directory to my $PATH environment variable</li>
</ol>
<div id="side-quest-install-htslib" class="section level4">
<h4>Side-quest: install HTSlib</h4>
<pre class="bash"><code>git clone https://github.com/samtools/htslib.git
cd htslib
git submodule update --init --recursive
autoreconf -i
./configure
make
make prefix=/net/snowwhite/home/beckandy/Tools/htslib install</code></pre>
</div>
</div>
<div id="getting-subject-information-for-1kgp" class="section level3">
<h3>Getting Subject Information for 1KGP</h3>
<p>From the <code>data</code> sub directory:</p>
<pre><code>wget ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/1000G_698_related_high_coverage.sequence.index
grep -v &#39;^##&#39; 1000G_698_related_high_coverage.sequence.index &gt; related_subj.tsv

wget ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/1000G_2504_high_coverage.sequence.index
grep -v &#39;^##&#39; 1000G_2504_high_coverage.sequence.index &gt; unrelated_subj.tsv

wget ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/1kGP.3202_samples.pedigree_info.txt

wget ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.superpopulations.tsv
wget ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv</code></pre>
<p>And then from the main directory of this project run</p>
<pre><code>Rscript code/combine_subj_info.R</code></pre>
<p>Using this dataset I randomly sampled a European male from the
unrelated sample (HG00260).</p>
</div>
<div id="first-look-at-a-vcf-file" class="section level3">
<h3>First look at a VCF file</h3>
<p>Let’s grab the genotypes for two subjects outside the
pseudo-autosomal region to verify that all these variants are
homozygous:</p>
<pre><code>bcftools view -t ^chrX:10001-2781479 -s HG00260,HG00136 -c1 \
/net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP0.minAC1.PASS/freeze9.1000g.chrX.filtered.gtonly.minDP0.minAC1.PASS.bcf |\
bcftools view -t ^chrX:155701382-156030895 |\
bcftools query -f &#39;%CHROM  %POS  %REF  %ALT{0} [ %GT]\n&#39; | tr -s &#39; &#39; &gt; data/chrX.tsv</code></pre>
<pre><code>bcftools view -t ^chrX:10001-2781479 -s HG00260,HG00136 -c1 \
/net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz |\
bcftools view -t ^chrX:155701382-156030895 |\
bcftools query -f &#39;%CHROM  %POS  %REF  %ALT{0} [ %GT]\n&#39; | tr -s &#39; &#39; &gt; data/chrX_phased.tsv</code></pre>
</div>
</div>
<div id="feb-2022-1" class="section level2">
<h2>23 Feb 2022</h2>
<div id="mask-the-vcf" class="section level3">
<h3>Mask the VCF</h3>
<p>The idea stemming from lab meeting was that the heterozygotes might
have been due to hard to sequence regions (repetitive regions, etc).
Let’s use a mask and see if that reduces the number of heterozygotes we
see in the two X chromosomes:</p>
<pre><code>zcat /net/wonderland/home/ycsi/Data/Ref/GRCh38/1kg_20160622_genome_mask/PilotMask/20160622.allChr.pilot_mask.bed.gz |\
awk &#39;{if($1 == &quot;chrX&quot;)print $0}&#39; &gt; data/mask/mask_pilot_chrX.bed

zcat /net/wonderland/home/ycsi/Data/Ref/GRCh38/1kg_20160622_genome_mask/StrictMask/20160622.allChr.mask.bed.gz |\
awk &#39;{if($1 == &quot;chrX&quot;)print $0}&#39; &gt; data/mask/mask_strict_chrX.bed</code></pre>
<p>NOTE: this version of the bed has un-masked regions -&gt; we want
intersections here</p>
<pre><code>bedtools intersect -a /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz -b /net/snowwhite/home/beckandy/research/phasing/data/mask_pilot_chrX.bed -wa -header |\
bcftools view -t ^chrX:10001-2781479 -s HG00260,HG00136 -c1 |\
bcftools view -t ^chrX:155701382-156030895 |\
bcftools query -f &#39;%CHROM  %POS  %REF  %ALT{0} [ %GT]\n&#39; | tr -s &#39; &#39; &gt; data/chrX_phased_mask.tsv</code></pre>
<p>And a different version of the mask (this version of the mask has
masked regions)</p>
<pre><code>zcat /net/wonderland/home/ycsi/Data/Ref/GRCh38/mdust.hg38.bed.gz |\
awk &#39;{if($1 == &quot;chrX&quot;)print $0}&#39; &gt; data/ref/mask_pilot_chrX_v2.bed</code></pre>
<pre><code>bedtools intersect -v -a /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz -b /net/snowwhite/home/beckandy/research/phasing/data/ref/mask_pilot_chrX_v2.bed -wa -header |\
bcftools view -t ^chrX:10001-2781479 -s HG00260,HG00136 -c1 -Ou |\
bcftools view -t ^chrX:155701382-156030895 -Ou |\
bcftools query -f &#39;%CHROM  %POS  %REF  %ALT{0} [ %GT]\n&#39; | tr -s &#39; &#39; &gt; data/chrX_phased_mask2.tsv</code></pre>
</div>
</div>
<div id="feb-2022-2" class="section level2">
<h2>28 Feb 2022</h2>
<p>Using the file <code>data/chrX_phased_mask2.tsv</code>, generate a
list of sites to keep in our VCF:</p>
<pre class="r"><code>library(tidyverse)
df &lt;- read_delim(&quot;data/chrX_phased_mask2.tsv&quot;, col_names = c(&quot;CHROM&quot;, &quot;POS&quot;, &quot;REF&quot;, &quot;ALT&quot;, &quot;G1&quot;, &quot;G2&quot;))
df2 &lt;- df %&gt;%
  filter(G1 %in% c(&quot;0|0&quot;, &quot;1|1&quot;),
         G2 %in% c(&quot;0|0&quot;, &quot;1|1&quot;))
df2 %&gt;%
  select(CHROM, POS) %&gt;%
  write_tsv(&quot;data/chrX_sites_28feb2022.tsv&quot;)</code></pre>
<p><strong>UPDATE: generate list of sites using bcftools
chain</strong></p>
<pre><code>bedtools intersect -v -a /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz -b /net/snowwhite/home/beckandy/research/phasing/data/ref/mask_pilot_chrX.bed -wa -header |\
bcftools view -t ^chrX:10001-2781479 -s HG00260,HG00136 -c1 -Ou |\
bcftools view -t ^chrX:155701382-156030895 -Ou |\
bcftools query -e &#39;GT=&quot;1|0&quot; || GT=&quot;0|1&quot;&#39; -f &#39;%CHROM \t%POS\n&#39; | tr -s &#39; &#39; &gt; data/chrX_sites_28feb2022_2.tsv</code></pre>
<pre><code>bcftools view -R /net/snowwhite/home/beckandy/research/phasing/data/chrX_sites_28feb2022.tsv -s HG00260,HG00136 -Ov /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz  &gt; data/chrX_test.vcf</code></pre>
<div id="running-shapeit4" class="section level3">
<h3>Running SHAPEIT4</h3>
<p>We need a genetic map, which we get from the SHAPEIT4 authors:</p>
<pre><code>wget https://github.com/odelaneau/shapeit4/raw/master/maps/genetic_maps.b38.tar.gz</code></pre>
<p>Also, our vcf file needs to be indexed:</p>
<pre><code># from the data dir
bcftools index chrX_test_fake.vcf.gz</code></pre>
<p>We also need an indexed reference panel -&gt; probably cheating if we
use 1000G?</p>
<pre><code>cp /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz data/chrX_ref.vcf.gz

bcftools index data/ref/chrX_ref.vcf.gz</code></pre>
<p>And now to phase our fake vcf:</p>
<pre><code>shapeit4.2 --input data/chrX_test_fake.vcf.gz \
  --map data/chrX.b38.gmap.gz \
  --reference data/first_attempt/chrX_ref.vcf.gz \
  --region chrX \
  --thread 10 \
  --log output/shapeit.log \
  --output data/first_attempt/chrX_phased_SHAPEIT.vcf.gz</code></pre>
</div>
<div id="installing-eagle-2.4" class="section level3">
<h3>Installing EAGLE 2.4</h3>
<p>Download what I believe are pre-compiled binaries</p>
<pre><code>wget https://storage.googleapis.com/broad-alkesgroup-public/Eagle/downloads/Eagle_v2.4.1.tar.gz
tar -zxvf Eagle_v2.4.1.tar.gz</code></pre>
<p>They have some instructions for preping a version of the 1KGP data
that no longer exist. Maybe worth processing my chr X reference data in
a similar manner?</p>
<p><a
href="https://alkesgroup.broadinstitute.org/Eagle/downloads/prep_1000G_hg38.txt"
class="uri">https://alkesgroup.broadinstitute.org/Eagle/downloads/prep_1000G_hg38.txt</a></p>
<p>Run this from the same directory with the refernce data
(<code>data/ref</code>):</p>
<pre><code>(bcftools view --no-version -h chrX_ref.vcf.gz |\
  grep -v &quot;^##contig=&lt;ID=[GNh]&quot; | sed &#39;s/^##contig=&lt;ID=MT/##contig=&lt;ID=chrM/;s/^##contig=&lt;ID=\([0-9XY]\)/##contig=&lt;ID=chr\1/&#39;; \
  bcftools view --no-version -H -c 2 chrX_ref.vcf.gz) | \
  bcftools norm --no-version -Ou -m -any | \
  bcftools norm --no-version -Ob -o chrX_ref.bcf -d none -f /net/snowwhite/home/beckandy/reference/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna &amp;&amp; \
  bcftools index -f chrX_ref.bcf</code></pre>
<p>And now we phase:</p>
<pre><code>eagle --vcfTarget /net/snowwhite/home/beckandy/research/phasing/data/first_attempt/chrX_test_fake.bcf \
  --vcfRef /net/snowwhite/home/beckandy/research/phasing/data/ref/chrX_ref.bcf \
  --geneticMapFile=/net/snowwhite/home/beckandy/software/Eagle_v2.4.1/tables/genetic_map_hg38_withX.txt.gz \
  --vcfOutFormat v \
  --chrom chrX \
  --outPrefix=eagle \
  --numThreads 10</code></pre>
</div>
</div>
<div id="march-2022" class="section level2">
<h2>2 March 2022</h2>
<div id="mask-vcf" class="section level3">
<h3>Mask VCF</h3>
<p><strong>NOTE: THIS CHANGED LATER IN APRIL</strong></p>
<p>Instead of running this 100 times, let’s just mask one copy of the X
chromosome (also removing the PAR)</p>
<pre><code>bedtools intersect -v -a /net/topmed8/working/call_sets/freeze9/release/subset/1000g/minDP10.minAC1.PASS.phased/freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz -b /net/snowwhite/home/beckandy/research/phasing/data/ref/mask_pilot_chrX_v2.bed -wa -header |\
bcftools view -t ^chrX:10001-2781479 -Ou |\
bcftools view -t ^chrX:155701382-156030895 -Ob &gt; data/ref/chrX_nonPAR.bcf</code></pre>
</div>
</div>
<div id="march-2022-1" class="section level2">
<h2>14 March 2022</h2>
<p>Test the approach: generate fake diploid, phase using entire sample
-&gt; should phase perfectly</p>
<p>EAGLE: phase confidence = 0.99995</p>
<p>Phased perfectly</p>
</div>
<div id="march-2022-2" class="section level2">
<h2>22 March 2022</h2>
<ul>
<li>Script to run pseudo-diploid analyses:
<code>code/batch_diploid_analysis.sh</code>
<ul>
<li>Wishlist: remove hard-coding of output directories (have it make
based on date)</li>
</ul></li>
<li>Results analysis:
<code>analysis/first_phasing_results.Rmd</code></li>
<li>Attempt at understanding how vcftools –diff-switch works:
<code>analysis/shift_error_ex.Rmd</code></li>
</ul>
<p><strong>Installing whatshap</strong></p>
<pre class="bash"><code># create conda env
conda create --name whatshap
# activate environment
conda activate whatshap
# install whatshap
conda install whatshap nomkl</code></pre>
</div>
<div id="april-2022" class="section level2">
<h2>11 April 2022</h2>
<ul>
<li>Annotation of switches with CpG status and 3mer motif:
<code>batch_annotate_switch.sh</code></li>
<li>Get background CpG and 3mer rates: <code>background_dimer.py</code>
and <code>background_3mer.py</code></li>
<li>Sampled 200 pairs of AFR subjects:
<code>sample_pairs_afr.csv</code></li>
</ul>
</div>
<div id="april-2022-1" class="section level2">
<h2>18 April 2022</h2>
<p><strong>NOTE:</strong> Some of this was later re-done using a masked
reference genome</p>
<div id="getting-gc-content-in-windows" class="section level3">
<h3>Getting GC Content in Windows</h3>
<pre class="bash"><code>curl -s http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes | \
  grep -Ev &quot;_|Y|M&quot; &gt; data/hg38.chrom.sizes
    
bedtools makewindows -g data/ref/hg38.chrom.sizes -w 1000000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.1000kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrom.sizes -w 5000000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.5000kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrom.sizes -w 100000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.100kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrom.sizes -w 10000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.10kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrom.sizes -w 1000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.1kb.sorted.bed

# GC content
bedtools nuc -fi data/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -bed data/ref/genome.1kb.sorted.bed | grep -E &#39;#|chrX&#39; &gt; data/ref/gc1kb_X_only.bed

bedtools nuc -fi data/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -bed data/ref/genome.10kb.sorted.bed &gt; data/ref/gc10kb.bed

bedtools nuc -fi data/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -bed data/ref/genome.100kb.sorted.bed &gt; data/ref/gc100kb.bed</code></pre>
</div>
</div>
<div id="april-2022-2" class="section level2">
<h2>19 April 2022</h2>
<div id="masked-vcf" class="section level3">
<h3>Masked VCF</h3>
<p>Re-did the step from March for creating the masked VCF, BUT we’ve
decided to use a different mask file (1KGP pilot)</p>
<pre><code> </code></pre>
</div>
</div>
<div id="april-2022-3" class="section level2">
<h2>20 April 2022</h2>
<div id="masked-reference-genome" class="section level3">
<h3>Masked reference genome</h3>
<p>From the data/ref directory:</p>
<pre><code>wget https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.chrom.sizes
grep &#39;chrX&#39; hg38.chrom.sizes &gt; hg38.chrX.sizes

wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz
gunzip GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz
fastaindex -f GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -i GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.in
fastafetch -f GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -i GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.in -q &quot;chrX&quot; &gt; chrX.fasta</code></pre>
<p>First we need to get the complement of the pilot mask:</p>
<pre><code>bedtools complement -i data/mask/mask_pilot_chrX.bed -g data/ref/hg38.chrX.sizes | bedtools sort &gt; data/mask/mask_pilot_fasta.bed</code></pre>
<p>Then we apply the mask to the fasta:</p>
<pre><code>bedtools maskfasta -fi data/ref/chrX.fasta -bed data/mask/mask_pilot_fasta.bed -fo data/ref/chrX_mask.fasta
samtools faidx data/ref/chrX_mask.fasta</code></pre>
<p>Re-make window files using reduced hg38.chrX.sizes file:</p>
<pre><code>grep &#39;chrX&#39; data/ref/hg38.chrom.sizes &gt; data/ref/hg38.chrX.sizes
bedtools makewindows -g data/ref/hg38.chrX.sizes -w 100000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.100kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrX.sizes -w 10000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.10kb.sorted.bed

bedtools makewindows -g data/ref/hg38.chrX.sizes -w 1000 | sort -k 1,1V -k2,2n &gt; data/ref/genome.1kb.sorted.bed</code></pre>
<p>And finally get GC content in windows:</p>
<pre><code>bedtools nuc -fi data/ref/chrX_mask.fasta -bed data/ref/genome.1kb.sorted.bed | grep -E &#39;#|chrX&#39; &gt; data/gc1kb_X_only.bed

bedtools nuc -fi data/ref/chrX_mask.fasta -bed data/ref/genome.10kb.sorted.bed &gt; data/gc10kb_pilot.bed

bedtools nuc -fi data/ref/chrX_mask.fasta -bed data/ref/genome.100kb.sorted.bed &gt; data/gc100kb_pilot.bed

# unmasked

bedtools nuc -fi data/ref/chrX.fasta -bed data/ref/genome.1kb.sorted.bed | grep -E &#39;#|chrX&#39; &gt; data/gc1kb_noMask.bed

bedtools nuc -fi data/ref/chrX.fasta -bed data/ref/genome.10kb.sorted.bed &gt; data/gc10kb_noMask.bed

bedtools nuc -fi data/ref/chrX.fasta -bed data/ref/genome.100kb.sorted.bed &gt; data/gc100kb_noMask.bed</code></pre>
</div>
</div>
<div id="may-2022" class="section level2">
<h2>23 May 2022</h2>
<p>Getting methylation maps from <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1127119">GEO:
GSM1127119</a></p>
<pre><code>wget https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1127nnn/GSM1127119/suppl/GSM1127119_UCSF-UBC.Testis_Spermatozoa_Primary_Cells.Bisulfite-Seq.390ATA.wig.gz

gunzip GSM1127119_UCSF-UBC.Testis_Spermatozoa_Primary_Cells.Bisulfite-Seq.390ATA.wig.gz

fetchChromSizes hg19 &gt; hg19.chrom.sizes
wigToBigWig GSM1127119_UCSF-UBC.Testis_Spermatozoa_Primary_Cells.Bisulfite-Seq.390ATA.wig hg19.chrom.sizes methylation.bw

bigWigToBedGraph methylation.bw methylation.bed 

wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
gunzip hg19ToHg38.over.chain.gz

liftOver methylation.bed hg19ToHg38.over.chain methylation_lo.bed unlifted.bed

grep &#39;chrX&#39; methylation_lo.bed | sort -k2,2n &gt; methylation_chrX.bed</code></pre>
</div>
<div id="june-2022" class="section level2">
<h2>8 June 2022</h2>
<ul>
<li>Created a script to make bed file with CpG locations
(<code>code/cpg_bed.py</code>)</li>
<li>Need to filter VCF to CpG</li>
</ul>
<p><code>bedtools intersect -a file.vcf -b cpg_pos.bed -wa -header | bcftools query -f "%CHROM\t%POS\t%REF\n"</code></p>
<p>Get CpG Het COunts:</p>
<pre><code>for i in `seq 1 300`; do
echo $i
bcftools view -e &#39;GT=&quot;1|1&quot; | GT=&quot;0|0&quot;&#39; andy_input_${i}_truth.vcf | bedtools intersect -a - -b ../../../../data/cpg_pos.bed -wa -header | bcftools query -f &quot;[%GT]\n&quot; | wc -l &gt;&gt; cpg_hets.txt
done</code></pre>
</div>
<div id="september-2022" class="section level2">
<h2>28 September 2022</h2>
<ul>
<li>Testing the effect of changing parameters
<ul>
<li>shapeit4 -&gt; does the –sequencing option reduce the number of
errors?</li>
<li>BEAGLE -&gt; Do we see better results when we don’t use the genetic
map?</li>
</ul></li>
</ul>
<p><code>batch_parameter_test.sh</code> generates the results</p>
</div>
<div id="march-2023" class="section level2">
<h2>MARCH 2023</h2>
<ul>
<li>Compare results using 1KGP datasets from
<ul>
<li>Internal (called against TOPMed)</li>
<li>gnomAD</li>
<li>NYGC (the 1KGP release)</li>
</ul></li>
</ul>
</div>
<div id="april-2023" class="section level2">
<h2>APRIL 2023</h2>
<ul>
<li>Installed SHAPEIT 5 -&gt; re-run pipeline, using samples from all
AFR, EUR individuals (i.e. not just the subset available in gnomAD)</li>
</ul>
</div>
<div id="august-2023" class="section level2">
<h2>August 2023</h2>
<div id="getting-reference-data" class="section level3">
<h3>Getting reference data</h3>
<p>Now our goal is to try and regress phasing errors on various measures
along the genome.</p>
<div id="reference-genome" class="section level4">
<h4>Reference Genome</h4>
<pre><code>gunzip -c GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz &gt; GCA_000001405.15_GRCh38_no_alt_analysis_set.fna
samtools faidx GCA_000001405.15_GRCh38_no_alt_analysis_set.fna chrX &gt; chrX_ref.fna
samtools faidx chrX_ref.fna
wget https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.chrom.sizes
bedtools complement -i ref/20160622.chrX.pilot_mask.bed -g ref/hg38.chrom.sizes | grep &#39;chrX&#39; &gt; inverse_pilot.bed
bedtools maskfasta -fi chrX_ref.fna -bed inverse_pilot.bed -fo chrX_ref_pilot.fna</code></pre>
</div>
<div id="gc-content" class="section level4">
<h4>GC content</h4>
<p>We can vary the size of the windows here. Initially we’ll look at GC
content in 10kb windows.</p>
<pre><code>bedtools nuc -fi chrX_ref_pilot.fna -bed ref/genome.10kb.sorted.bed &gt; chrX_gc_10kb.bed</code></pre>
</div>
<div id="recombination-rates" class="section level4">
<h4>Recombination Rates</h4>
<p>We’ll first take a look at data from <a
href="https://doi.org/10.1126/science.aau1043"
class="uri">https://doi.org/10.1126/science.aau1043</a></p>
<p>See <code>README.md</code> in the <code>data/recomb</code> folder for
details on how this data was processed.</p>
</div>
</div>
</div>
<div id="october-2023" class="section level2">
<h2>October 2023</h2>
<p>Let’s ensure that our construction of the masked vcf file is
correct.</p>
<div id="step-1-remove-par-and-apply-mask" class="section level3">
<h3>Step 1: Remove PAR and apply mask</h3>
<p>run from the <code>data/1kgp</code> directory</p>
<pre><code>bcftools view -Ov --types snps -S unrel_ids.txt 1kGP_high_coverage_Illumina.chrX.filtered.SNV_INDEL_SV_phased_panel.v2.vcf.gz |\
bcftools view -Ov -t ^chrX:10001-2781479 |\
bcftools view -Ov -t ^chrX:155701382-156030895 |\
bedtools intersect -a - -b 20160622.chrX.pilot_mask.bed -wa -header |\
bcftools view -Ob &gt; 1kGP_nonPAR_mask.bcf</code></pre>
</div>
<div id="step-2-remove-singletons" class="section level3">
<h3>Step 2: Remove singletons</h3>
<pre><code>bcftools view -Ov 1kGP_nonPAR_mask.bcf | vcftools --singletons --vcf -
awk &#39;{if($1 == &quot;chrX&quot;)print($1&quot;\t&quot;$2)}&#39; out.singletons &gt; exclude.txt
bcftools view -Ob -T ^exclude.txt 1kGP_nonPAR_mask.bcf &gt; 1kGP_nonPAR_mask_noSing.bcf</code></pre>
</div>
</div>
<div id="getting-pd-sizes" class="section level2">
<h2>Getting PD sizes</h2>
<p>computing the distance between last and first heterozygote in each
pseudo-diploid</p>
<pre><code>for i in `seq 1 700`; do
begin=$(head -1 &quot;pair_${i}_het_loc.txt&quot; | awk &#39;{print($2)}&#39;)
end=$(tail -1 &quot;pair_${n}_het_loc.txt&quot; | awk &#39;{print($2)}&#39;)
(( dist = end - begin ))
echo &quot;${i}\t${dist}&quot; &gt;&gt; phased_size.txt
done</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Detroit
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5      httr_1.4.6       cli_3.6.1        knitr_1.42      
 [5] rlang_1.1.2      xfun_0.39        stringi_1.8.2    processx_3.8.1  
 [9] promises_1.2.0.1 jsonlite_1.8.7   glue_1.6.2       rprojroot_2.0.3 
[13] git2r_0.32.0     htmltools_0.5.6  httpuv_1.6.11    ps_1.7.5        
[17] sass_0.4.6       fansi_1.0.5      rmarkdown_2.22   jquerylib_0.1.4 
[21] tibble_3.2.1     evaluate_0.21    fastmap_1.1.1    yaml_2.3.7      
[25] lifecycle_1.0.4  whisker_0.4.1    stringr_1.5.1    compiler_4.3.2  
[29] fs_1.6.2         pkgconfig_2.0.3  Rcpp_1.0.11      rstudioapi_0.14 
[33] later_1.3.1      digest_0.6.33    R6_2.5.1         utf8_1.2.4      
[37] pillar_1.9.0     callr_3.7.3      magrittr_2.0.3   bslib_0.5.0     
[41] tools_4.3.2      cachem_1.0.8     getPass_0.2-2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
