<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2022-04-20" />

<title>masked_diploid_results</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">masked_diploid_results</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2022-04-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>phasing/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220221code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220221)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220221code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220221)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/eagle/annotated/
</td>
<td style="text-align:left;">
output/19april22_switch_errors/switch_errors/eagle/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/shapeit/annotated/
</td>
<td style="text-align:left;">
output/19april22_switch_errors/switch_errors/shapeit/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/vcf_n_sites/
</td>
<td style="text-align:left;">
output/19april22_switch_errors/vcf_n_sites
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/whatshap/
</td>
<td style="text-align:left;">
output/19april22_switch_errors/whatshap
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/background_rates/dimer.csv
</td>
<td style="text-align:left;">
output/background_rates/dimer.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/background_rates/3mer.csv
</td>
<td style="text-align:left;">
output/background_rates/3mer.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/data/sample_pairs_all.csv
</td>
<td style="text-align:left;">
data/sample_pairs_all.csv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong0add4ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 0add4ba
</a>
</p>
</div>
<div id="strongRepositoryversionstrong0add4ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 0add4ba.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/first_attempt/
    Ignored:    data/ref/
    Ignored:    data/shapeit/
    Ignored:    output/11april22_switch_errors/
    Ignored:    output/14mar22_switch_errors/
    Ignored:    output/16mar22_switch_errors/
    Ignored:    output/18mar22_switch_errors/
    Ignored:    output/19april22_switch_errors/
    Ignored:    output/background_rates/
    Ignored:    output/switch_error_test/
    Ignored:    sandbox/

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/first_phase_test.Rmd
    Untracked:  analysis/first_phasing_results.Rmd
    Untracked:  analysis/gc_content.Rmd
    Untracked:  analysis/shift_error_ex.Rmd
    Untracked:  code/background_3mer.py
    Untracked:  code/background_dimer.py

Unstaged changes:
    Modified:   analysis/build_a_vcf.Rmd
    Modified:   analysis/captains_log.Rmd
    Modified:   analysis/data_exploration.Rmd
    Modified:   code/batch_annotate_switch.sh
    Modified:   code/batch_diploid_analysis.sh
    Modified:   code/sample_X_pairs.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/masked_diploid_results.Rmd</code>) and HTML
(<code>docs/masked_diploid_results.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
0add4ba
</td>
<td>
Andy Beck
</td>
<td>
2022-05-02
</td>
<td>
wflow_publish("analysis/masked_diploid_results.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.5     ✓ purrr   0.3.4
✓ tibble  3.1.6     ✓ dplyr   1.0.8
✓ tidyr   1.2.0     ✓ stringr 1.4.0
✓ readr   2.1.2     ✓ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>eagle_switch_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/eagle/annotated/&quot;
shapeit_switch_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/shapeit/annotated/&quot;
num_sites_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/vcf_n_sites/&quot;
whatshap_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/whatshap/&quot;

pair_info_df &lt;- read_delim(&quot;data/sample_pairs_all.csv&quot;, col_names = c(&quot;POP&quot;, &quot;ID1&quot;, &quot;ID2&quot;))</code></pre>
<pre><code>Rows: 300 Columns: 3</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): POP, ID1, ID2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pair_info_df$SP &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))

gc_content_1kb &lt;- read_tsv(&quot;data/gc1kb_X_only.bed&quot;)</code></pre>
<pre><code>Rows: 156041 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (1): #1_usercol
dbl (11): 2_usercol, 3_usercol, 4_pct_at, 5_pct_gc, 6_num_A, 7_num_C, 8_num_...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>colnames(gc_content_1kb) &lt;- c(&quot;CHR&quot;, &quot;START&quot;, &quot;END&quot;, &quot;AT&quot;, &quot;GC&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;TOTAL&quot;, &quot;OTHER&quot;, &quot;LENGTH&quot;)
gc_content_1kb  &lt;- gc_content_1kb %&gt;%
  mutate(bin_id = (START / 1000) + 1)

background_dimer &lt;- read_csv(&quot;/net/snowwhite/home/beckandy/research/phasing/output/background_rates/dimer.csv&quot;)</code></pre>
<pre><code>Rows: 16 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): Nucs
dbl (1): N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>background_3mer &lt;- read_csv(&quot;/net/snowwhite/home/beckandy/research/phasing/output/background_rates/3mer.csv&quot;)</code></pre>
<pre><code>Rows: 64 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): Nucs
dbl (1): N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We’ve refined our analyses in a few ways since the first run-through
of this pseudo-diploid analysis. First let’s list out everything I’ve
done to get to this point:</p>
<ol style="list-style-type: decimal">
<li>Subset the reference genome to just the X chromosome using
<code>fastafetch</code></li>
<li>Download pilot mask from 1KGP along with chromosome lengths for hg38
<ul>
<li>Take complement of mask with bedtools to get file with masked
regions (as opposed to included regions)</li>
</ul></li>
<li>Mask both the reference fasta (<code>bedtools maskfasta</code>) and
the 1KGP X chromosome vcf
<ul>
<li><code>freeze9.1000g.chrX.filtered.gtonly.minDP10.minAC1.PASS.phased.vcf.gz</code></li>
</ul></li>
<li>Make window files (<code>bedtools makewindows</code>)</li>
<li>Get GC content in windows (<code>bedtools nuc</code>) using masked
fasta</li>
<li>Remove PAR from X chromosome</li>
</ol>
<p>Using these masked reference and variant files, we proceeded to do
our pseudo-diploid generation on 100 EUR and 200 AFR pairs:</p>
<ol style="list-style-type: decimal">
<li>Sample a sub-population from 1KGP super-population (EUR or AFR)</li>
<li>Sample 2 male samples from sub-population</li>
<li>Remove remaining heterozygous sites in either subject (should not be
heterozygous in non-PAR)</li>
<li>Create truth VCF with haploids pasted together, and test vcf with
sites in haploids shuffled</li>
<li>Create reference VCF by removing two sampled subjects</li>
<li>Phase the test VCF using EAGLE and SHAPEIT</li>
<li>Compare phasing results and truth using <code>vcftools</code> and
<code>whatshap</code></li>
</ol>
<p>With those preliminaries out of the way, let’s take a look at these
results.</p>
</div>
<div id="example-of-a-single-pseudodiploid" class="section level2">
<h2>Example of a single pseudodiploid</h2>
<p>For a first pass, let’s take a look at the information available for
a single pseudo-diploid:</p>
<pre class="r"><code>pair_id &lt;- 1
switch_err_eagle &lt;- read_csv(paste0(eagle_switch_dir, &quot;switch_&quot;, pair_id, &quot;.csv&quot;))</code></pre>
<pre><code>Rows: 965 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): motif_start, end_motif
dbl (4): pos_start, pos_end, cpg_start, cpg_end

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>switch_err_shapeit &lt;- read_csv(paste0(shapeit_switch_dir, &quot;switch_&quot;, pair_id, &quot;.csv&quot;))</code></pre>
<pre><code>Rows: 961 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): motif_start, end_motif
dbl (4): pos_start, pos_end, cpg_start, cpg_end

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>whatshap_eagle &lt;- read_tsv(paste0(whatshap_dir, &quot;/eagle/eval_&quot;, pair_id, &quot;.tsv&quot;))</code></pre>
<pre><code>Rows: 1 Columns: 28
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (8): #sample, chromosome, dataset_name0, dataset_name1, file_name0, fil...
dbl (20): intersection_blocks, covered_variants, all_assessed_pairs, all_swi...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>whatshap_shapeit &lt;- read_tsv(paste0(whatshap_dir, &quot;/shapeit/eval_&quot;, pair_id, &quot;.tsv&quot;))</code></pre>
<pre><code>Rows: 1 Columns: 28
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (8): #sample, chromosome, dataset_name0, dataset_name1, file_name0, fil...
dbl (20): intersection_blocks, covered_variants, all_assessed_pairs, all_swi...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>From the above we can get the raw number of total switches based on
the dimension of the switch error data frames; in this example, we see
that EAGLE has 965 switches and SHAPEIT has 961 switches.</p>
<p>From the whatshap output, we also get a breakdown between how many of
the total switches are flips. For example, the breakdown between
switches and flips (switches/flips) in EAGLE is 289/338, and
correspondingly in SHAPEIT we see 211/375. What we would really like is
to know the locations of the flips, which we unfortunatly don’t have
from neither <code>vcftools</code> nor <code>whatshap</code>.
Fortunately I believe the little chunk of code below correctly counts
flips in the <code>vcftools</code> output:</p>
<pre class="r"><code>count &lt;- 0
in_progress &lt;- FALSE
for(i in 1:(length(switch_err_eagle$pos_start)-1)){
  if(switch_err_eagle$pos_end[i] == switch_err_eagle$pos_start[i+1]){
    if(!in_progress){
      count &lt;- count+1
      in_progress &lt;- TRUE
    } else{
      in_progress &lt;- FALSE
    }
  } else{
    in_progress &lt;- FALSE
  }
}
print(count)</code></pre>
<pre><code>[1] 338</code></pre>
<p>Let’s wrap this in a function and get a list of flip positions:</p>
<pre class="r"><code>get_flip_pos &lt;- function(df){
  flip_list &lt;- c()
  in_progress &lt;- FALSE
  for(i in 1:(length(df$pos_start)-1)){
    if(df$pos_end[i] == df$pos_start[i+1]){
      if(!in_progress){
        in_progress &lt;- TRUE
        flip_list &lt;- c(flip_list, df$pos_end[i])
      } else{
        in_progress &lt;- FALSE
      }
    } else{
      in_progress &lt;- FALSE
    }
  }
  return(flip_list)
}

flip_pos_eagle &lt;- get_flip_pos(switch_err_eagle)
flip_pos_shapeit &lt;- get_flip_pos(switch_err_shapeit)</code></pre>
<p>Just to satisfy my curiosity, let’s see how many of the flip
positions are shared between the methods:</p>
<pre class="r"><code>intersect(flip_pos_eagle, flip_pos_shapeit) %&gt;% length()</code></pre>
<pre><code>[1] 191</code></pre>
<p>And moving on, let’s go ahead and annotate each switch in the
<code>vcftools</code> output with whether or not it is a flip:</p>
<pre class="r"><code>switch_err_eagle$is_flip &lt;- switch_err_eagle$pos_start %in% flip_pos_eagle
switch_err_shapeit$is_flip &lt;- switch_err_shapeit$pos_start %in% flip_pos_shapeit</code></pre>
<p>Alright, one last thing we’ll look at for this pair is the proportion
of all switches and flips are CpGs and contrast this with the background
rate of dimers. For reference, the distribution of dimers in the X
chromosome is:</p>
<pre class="r"><code>background_dimer</code></pre>
<pre><code># A tibble: 16 × 2
   Nucs         N
   &lt;chr&gt;    &lt;dbl&gt;
 1 AA    14628685
 2 AC     7264337
 3 AG     9841023
 4 AT    11735166
 5 CA    10391331
 6 CC     6920168
 7 CG     1133731
 8 CT     9846580
 9 GA     8486790
10 GC     5619375
11 GG     6935465
12 GT     7287386
13 TA     9961129
14 TC     8487812
15 TG    10420031
16 TT    14715291</code></pre>
<p>and by summing up the number of CG and GC sites and dividing by the
total we see that the proportion of sites that are CpGs is 0.0470029
(around 4.7%)</p>
<pre class="r"><code>background_cpg &lt;- (5619375 + 1133731) / sum(background_dimer$N)

switch_err_eagle %&gt;%
  filter(is_flip == TRUE) %&gt;%
  pull(cpg_start) %&gt;%
  table()</code></pre>
<pre><code>.
  0   1 
302  36 </code></pre>
<p>Here for EAGLE we see that about 10.6% of flips occur at CpG
sites.</p>
<pre class="r"><code>switch_err_shapeit %&gt;%
  filter(is_flip == TRUE) %&gt;%
  pull(cpg_start) %&gt;%
  table()</code></pre>
<pre><code>.
  0   1 
335  40 </code></pre>
<p>SHAPEIT also has ~ 10.6% of its flips at CpG. Now looking at overall
number of switches:</p>
<pre class="r"><code>switch_err_eagle %&gt;%
  pull(cpg_start) %&gt;%
  table()</code></pre>
<pre><code>.
  0   1 
886  79 </code></pre>
<pre class="r"><code>switch_err_shapeit %&gt;%
  pull(cpg_start) %&gt;%
  table()</code></pre>
<pre><code>.
  0   1 
872  89 </code></pre>
<p>8.2% and 9.3% of switches at CpG.</p>
<div id="gc-content" class="section level3">
<h3>GC Content</h3>
<pre class="r"><code>switch_err_eagle &lt;- switch_err_eagle %&gt;%
  mutate(bin_id = ceiling(pos_start / 1000))

switch_err_shapeit &lt;- switch_err_shapeit %&gt;%
  mutate(bin_id = ceiling(pos_start / 1000))

switch_err_eagle &lt;- switch_err_eagle %&gt;%
    left_join({gc_content_1kb %&gt;% select(bin_id, GC)}, by = &quot;bin_id&quot;)
  
switch_err_shapeit &lt;- switch_err_shapeit %&gt;%
    left_join({gc_content_1kb %&gt;% select(bin_id, GC)}, by = &quot;bin_id&quot;)</code></pre>
</div>
</div>
<div id="summary-statisitics-across-all-pseudo-diploids"
class="section level2">
<h2>Summary Statisitics Across All Pseudo-diploids</h2>
<p>Let’s grab:</p>
<ul>
<li>Number of heterozygous sites per diploid</li>
<li>EAGLE phase confidence</li>
<li>Total number of switch errors for each method</li>
<li>Number of flips for each method</li>
<li>Percent of flips and switches that were CpG site</li>
<li>Average percent GC in 1kb windows</li>
</ul>
<div id="pair-information" class="section level3">
<h3>Pair information</h3>
<pre class="r"><code>df_pair &lt;- read_csv(&quot;/net/snowwhite/home/beckandy/research/phasing/data/sample_pairs_all.csv&quot;,
                    col_names = c(&quot;pop&quot;, &quot;sub1&quot;, &quot;sub2&quot;), show_col_types = FALSE) %&gt;%
  rename(sub_pop = pop)
df_pair$pop &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))
df_pair$pair_id &lt;- 1:300</code></pre>
</div>
<div id="eagle-phase-confidence" class="section level3">
<h3>EAGLE Phase Confidence</h3>
<pre class="r"><code>input_cmd &lt;- &quot;grep \&quot;^FAKE001\&quot; /net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/slurm/pair.*.out | awk &#39;{print($1, \&quot;\\t\&quot;, $2)}&#39;&quot;

df_eagle_conf &lt;- vroom::vroom(pipe(input_cmd), delim = &quot;\t&quot;, col_names = c(&quot;f_name&quot;, &quot;phase_confidence&quot;)) %&gt;%
  rowwise() %&gt;%
  mutate(pair_id = as.numeric(str_split(f_name, &quot;\\.&quot;)[[1]][3])) %&gt;%
  select(pair_id, phase_confidence) %&gt;%
  arrange(pair_id)</code></pre>
<pre><code>Rows: 300 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): f_name
dbl (1): phase_confidence

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_eagle_conf$pop &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))</code></pre>
<p>Oh, let’s make a density plot, shall we?</p>
<pre class="r"><code>df_eagle_conf %&gt;%
  ggplot(aes(x = phase_confidence, colour = pop)) +
  geom_density() +
  ggtitle(&quot;EAGLE2 Phase Confidence&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-sites" class="section level3">
<h3>Number of sites</h3>
<pre class="r"><code>input_cmd &lt;- &quot;cat /net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/vcf_n_sites/*all*&quot;
df_pseudo_sites &lt;- vroom::vroom(pipe(input_cmd), delim = &quot;\t&quot;, col_names = c(&quot;pair_id&quot;, &quot;n_sites&quot;))  %&gt;%
  arrange(pair_id)</code></pre>
<pre><code>Rows: 300 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
dbl (2): pair_id, n_sites

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_pseudo_sites$pop &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))</code></pre>
<p>And just to plot things out as we go, let’s look at the density
stratified by super-pop:</p>
<pre class="r"><code>df_pseudo_sites %&gt;%
  ggplot(aes(x = n_sites, colour = pop)) +
  geom_density() +
  ggtitle(&quot;Number of Sites per Pseudo-diploid&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-heterozygous-sites" class="section level3">
<h3>Number of heterozygous sites</h3>
<pre class="r"><code>input_cmd &lt;-  &quot;cat /net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/vcf_n_sites/*hets*&quot;
df_pseudo_hets &lt;- vroom::vroom(pipe(input_cmd), delim = &quot;\t&quot;, col_names = c(&quot;pair_id&quot;, &quot;n_het_sites&quot;))  %&gt;%
  arrange(pair_id)</code></pre>
<pre><code>Rows: 300 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
dbl (2): pair_id, n_het_sites

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_pseudo_hets$pop &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))</code></pre>
<pre class="r"><code>df_pseudo_hets %&gt;%
  ggplot(aes(x = n_het_sites, colour = pop)) +
  geom_density() +
  xlab(&quot;Number of heterozygous sites&quot;) +
  ggtitle(&quot;Number of Heterozygous Sites per Pseudo-diploid&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="switches-and-flips" class="section level3">
<h3>Switches and Flips</h3>
<pre class="r"><code>switch_summary &lt;- function(pair_id, eagle_dir, shapeit_dir, gc_content_1kb, bin_size = 1000){
  switch_err_eagle &lt;- read_csv(paste0(eagle_dir, &quot;switch_&quot;, pair_id, &quot;.csv&quot;), show_col_types = FALSE) %&gt;%
    mutate(bin_id = ceiling(pos_start / bin_size))
  switch_err_shapeit &lt;- read_csv(paste0(shapeit_dir, &quot;switch_&quot;, pair_id, &quot;.csv&quot;), show_col_types = FALSE) %&gt;%
    mutate(bin_id = ceiling(pos_start / bin_size))
  
  switch_err_eagle &lt;- switch_err_eagle %&gt;%
    left_join({gc_content_1kb %&gt;% select(bin_id, GC)}, by = &quot;bin_id&quot;)
  
  switch_err_shapeit &lt;- switch_err_shapeit %&gt;%
    left_join({gc_content_1kb %&gt;% select(bin_id, GC)}, by = &quot;bin_id&quot;)
  
  # get positions of flips
  flip_pos_eagle &lt;- get_flip_pos(switch_err_eagle)
  flip_pos_shapeit &lt;- get_flip_pos(switch_err_shapeit)
  
  # Assign switches flip status
  switch_err_eagle$is_flip &lt;- switch_err_eagle$pos_start %in% flip_pos_eagle
  switch_err_shapeit$is_flip &lt;- switch_err_shapeit$pos_start %in% flip_pos_shapeit
  
  # stats we want to pull
  n_switch_eagle &lt;- length(switch_err_eagle$pos_start)
  n_switch_shapeit &lt;- length(switch_err_shapeit$pos_start)
  n_flip_eagle &lt;- sum(switch_err_eagle$is_flip)
  n_flip_shapeit &lt;- sum(switch_err_shapeit$is_flip)
  n_switch_cpg_eagle &lt;- switch_err_eagle %&gt;%
    filter(cpg_start == 1) %&gt;%
    pull(is_flip) %&gt;%
    length()
  n_flip_cpg_eagle &lt;- switch_err_eagle %&gt;%
    filter(cpg_start == 1) %&gt;%
    pull(is_flip) %&gt;%
    sum()
  n_switch_cpg_shapeit &lt;- switch_err_shapeit %&gt;%
    filter(cpg_start == 1) %&gt;%
    pull(is_flip) %&gt;%
    length()
  n_flip_cpg_shapeit &lt;- switch_err_shapeit %&gt;%
    filter(cpg_start == 1) %&gt;%
    pull(is_flip) %&gt;%
    sum()
  mean_gc_switch_eagle &lt;- mean(switch_err_eagle$GC)
  mean_gc_switch_shapeit &lt;- mean(switch_err_shapeit$GC)
  
  return(data.frame(pair_id = pair_id,
                    n_switch_eagle = n_switch_eagle,
                    n_switch_shapeit = n_switch_shapeit,
                    n_flip_eagle = n_flip_eagle,
                    n_flip_shapeit = n_flip_shapeit,
                    n_switch_cpg_eagle = n_switch_cpg_eagle,
                    n_flip_cpg_eagle = n_flip_cpg_eagle,
                    n_switch_cpg_shapeit = n_switch_cpg_shapeit,
                    n_flip_cpg_shapeit = n_flip_cpg_shapeit,
                    mean_gc_switch_eagle = mean_gc_switch_eagle,
                    mean_gc_switch_shapeit = mean_gc_switch_shapeit
                    ))
}</code></pre>
<p>And now let’s get the rows for all 300 pseudo-diploids:</p>
<pre class="r"><code>switch_df &lt;- lapply(c(1:300), 
                    function(x){
                      switch_summary(x, eagle_switch_dir, shapeit_switch_dir, gc_content_1kb)
                      }) %&gt;%
  bind_rows()

switch_df$pop &lt;- c(rep(&quot;EUR&quot;, 100), rep(&quot;AFR&quot;, 200))</code></pre>
</div>
<div id="smash-everything-together-into-a-single-data-frame"
class="section level3">
<h3>Smash everything together into a single data frame</h3>
<pre class="r"><code>df &lt;- full_join(switch_df, df_pair, by = c(&quot;pair_id&quot;, &quot;pop&quot;)) %&gt;%
  full_join(df_eagle_conf, by = c(&quot;pair_id&quot;, &quot;pop&quot;)) %&gt;%
  full_join(df_pseudo_sites, by = c(&quot;pair_id&quot;, &quot;pop&quot;)) %&gt;%
  full_join(df_pseudo_hets, by = c(&quot;pair_id&quot;, &quot;pop&quot;))</code></pre>
<p>And now: we make graphs</p>
<p><strong>Number of switch error</strong></p>
<pre class="r"><code>df %&gt;%
  ggplot(aes(x = n_switch_eagle, y = n_switch_shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Switch Errors: EAGLE&quot;) +
  ylab(&quot;Switch Errors: SHAPEIT&quot;) +
  ggtitle(&quot;Switch Errors by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  ggplot(aes(x = n_switch_eagle, y = n_switch_shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Switch Errors: EAGLE&quot;) +
  ylab(&quot;Switch Errors: SHAPEIT&quot;) +
  ggtitle(&quot;Switch Errors by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Number of switch error - % heterozygosity</strong></p>
<pre class="r"><code>df %&gt;%
  mutate(pct_het = n_het_sites / n_sites) %&gt;%
  ggplot(aes(x = n_switch_eagle, y = n_switch_shapeit, colour = pct_het)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Switch Errors: EAGLE&quot;) +
  ylab(&quot;Switch Errors: SHAPEIT&quot;) +
  ggtitle(&quot;Switch Errors by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Number of flip error</strong></p>
<pre class="r"><code>df %&gt;%
  ggplot(aes(x = n_flip_eagle, y = n_flip_shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Flips: EAGLE&quot;) +
  ylab(&quot;Flips: SHAPEIT&quot;) +
  ggtitle(&quot;Flips by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(prop1 = n_switch_cpg_eagle / n_switch_eagle,
         prop2 = n_switch_cpg_shapeit / n_switch_shapeit) %&gt;%
  ggplot(aes(x = prop1, y = prop2, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Proportion Switches CpG: EAGLE&quot;) +
  ylab(&quot;Proportion Switches CpG: SHAPEIT&quot;) +
  ggtitle(&quot;Proportion Switches CpG by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Density version of above plot:</p>
<pre class="r"><code>df %&gt;%
  mutate(eagle = n_switch_cpg_eagle / n_switch_eagle,
         shapeit = n_switch_cpg_shapeit / n_switch_shapeit) %&gt;%
  select(pair_id, pop, eagle, shapeit) %&gt;%
  pivot_longer(eagle:shapeit, &quot;method&quot;, values_to = &quot;switch_enrich&quot;) %&gt;%
  ggplot(aes(x = switch_enrich, colour = method)) +
  geom_density() +
  ggtitle(&quot;Total Error Proportion CpG&quot;) + 
  xlab(&quot;CpG Error Proportion&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(prop1 = n_flip_cpg_eagle / n_flip_eagle,
         prop2 = n_flip_cpg_shapeit / n_flip_shapeit) %&gt;%
  ggplot(aes(x = prop1, y = prop2, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;Proportion Flips CpG: EAGLE&quot;) +
  ylab(&quot;Proportion Flips CpG: SHAPEIT&quot;) +
  ggtitle(&quot;Proportion Flips CpG by Method&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(eagle = n_flip_cpg_eagle / n_flip_eagle,
         shapeit = n_flip_cpg_shapeit / n_flip_shapeit) %&gt;%
  select(pair_id, pop, eagle, shapeit) %&gt;%
  pivot_longer(eagle:shapeit, &quot;method&quot;, values_to = &quot;flip_enrich&quot;) %&gt;%
  ggplot(aes(x = flip_enrich, colour = method)) +
  geom_density() +
  ggtitle(&quot;Flip Proportion CpG&quot;) + 
  xlab(&quot;CpG Flip Proportion&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="new-stuff-post-meeting" class="section level3">
<h3>New stuff post meeting</h3>
<p>Ratio of proportion errors to background CpG rate</p>
<pre class="r"><code>df %&gt;%
  mutate(enrich_eagle = (n_switch_cpg_eagle/n_switch_eagle) / background_cpg,
         enrich_shapeit = (n_switch_cpg_shapeit/n_switch_shapeit) / background_cpg) %&gt;%
  ggplot(aes(x = enrich_eagle, y = enrich_shapeit, color = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  ggtitle(&quot;Total Error CpG Enrichment&quot;) +
  xlab(&quot;EAGLE enrichment&quot;) +
  ylab(&quot;SHAPEIT enrichment&quot;) </code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(eagle = (n_switch_cpg_eagle/n_switch_eagle) / background_cpg,
         shapeit = (n_switch_cpg_shapeit/n_switch_shapeit) / background_cpg) %&gt;%
  select(pair_id, pop, eagle, shapeit) %&gt;%
  pivot_longer(eagle:shapeit, &quot;method&quot;, values_to = &quot;flip_enrich&quot;) %&gt;%
  ggplot(aes(x = flip_enrich, colour = method)) +
  geom_density() +
  ggtitle(&quot;Flip Proportion CpG&quot;) + 
  xlab(&quot;CpG Flip Proportion&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>CpG Flip Enrichment</p>
<pre class="r"><code>df %&gt;%
  mutate(enrich_eagle = (n_flip_cpg_eagle/n_flip_eagle) / background_cpg,
         enrich_shapeit = (n_flip_cpg_shapeit/n_flip_shapeit) / background_cpg) %&gt;%
  ggplot(aes(x = enrich_eagle, y = enrich_shapeit, color = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  xlab(&quot;EAGLE enrichment&quot;) +
  ylab(&quot;SHAPEIT enrichment&quot;) +
  ggtitle(&quot;Flip CpG Enrichment&quot;)</code></pre>
<p><img src="figure/masked_diploid_results.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.8     purrr_0.3.4    
 [5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.6    ggplot2_3.3.5  
 [9] tidyverse_1.3.1 workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8       lubridate_1.8.0  getPass_0.2-2    ps_1.6.0        
 [5] assertthat_0.2.1 rprojroot_2.0.2  digest_0.6.29    utf8_1.2.2      
 [9] cellranger_1.1.0 R6_2.5.1         backports_1.4.1  reprex_2.0.1    
[13] evaluate_0.14    httr_1.4.2       highr_0.9        pillar_1.7.0    
[17] rlang_1.0.1      readxl_1.3.1     rstudioapi_0.13  whisker_0.4     
[21] callr_3.7.0      jquerylib_0.1.4  rmarkdown_2.11   labeling_0.4.2  
[25] bit_4.0.4        munsell_0.5.0    broom_0.8.0      compiler_4.2.0  
[29] httpuv_1.6.5     modelr_0.1.8     xfun_0.29        pkgconfig_2.0.3 
[33] htmltools_0.5.2  tidyselect_1.1.1 fansi_1.0.2      withr_2.4.3     
[37] crayon_1.4.2     tzdb_0.2.0       dbplyr_2.1.1     later_1.3.0     
[41] grid_4.2.0       jsonlite_1.7.3   gtable_0.3.0     lifecycle_1.0.1 
[45] DBI_1.1.2        git2r_0.29.0     magrittr_2.0.2   scales_1.1.1    
[49] vroom_1.5.7      cli_3.1.1        stringi_1.7.6    farver_2.1.0    
[53] fs_1.5.2         promises_1.2.0.1 xml2_1.3.3       bslib_0.3.1     
[57] ellipsis_0.3.2   generics_0.1.2   vctrs_0.3.8      tools_4.2.0     
[61] bit64_4.0.5      glue_1.6.1       hms_1.1.1        parallel_4.2.0  
[65] processx_3.5.2   fastmap_1.1.0    yaml_2.2.2       colorspace_2.0-2
[69] rvest_1.0.2      knitr_1.37       haven_2.4.3      sass_0.4.0      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
