<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2022-05-16" />

<title>methylation_explore</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">methylation_explore</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2022-05-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>phasing/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220221code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220221)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220221code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220221)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/eagle/annotated
</td>
<td style="text-align:left;">
output/19april22_switch_errors/switch_errors/eagle/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/shapeit/annotated
</td>
<td style="text-align:left;">
output/19april22_switch_errors/switch_errors/shapeit/annotated
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongc753b7c">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> c753b7c
</a>
</p>
</div>
<div id="strongRepositoryversionstrongc753b7c"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version c753b7c.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/first_attempt/
    Ignored:    data/ovary/
    Ignored:    data/ref/
    Ignored:    data/shapeit/
    Ignored:    output/11april22_switch_errors/
    Ignored:    output/14mar22_switch_errors/
    Ignored:    output/16mar22_switch_errors/
    Ignored:    output/18mar22_switch_errors/
    Ignored:    output/19april22_switch_errors/
    Ignored:    output/background_rates/
    Ignored:    output/switch_error_test/
    Ignored:    sandbox/

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/first_phase_test.Rmd
    Untracked:  analysis/first_phasing_results.Rmd
    Untracked:  analysis/gc_content.Rmd
    Untracked:  analysis/methylation_explore.Rmd
    Untracked:  analysis/shift_error_ex.Rmd
    Untracked:  code/add_meth.R
    Untracked:  code/background_3mer.py
    Untracked:  code/background_dimer.py
    Untracked:  code/batch_add_meth.sh
    Untracked:  plink.log

Unstaged changes:
    Modified:   analysis/build_a_vcf.Rmd
    Modified:   analysis/captains_log.Rmd
    Modified:   analysis/data_exploration.Rmd
    Modified:   analysis/masked_diploid_results.Rmd
    Modified:   code/batch_annotate_switch.sh
    Modified:   code/batch_diploid_analysis.sh
    Modified:   code/sample_X_pairs.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>File: <code>data/methylation_chrX.bed</code></p>
<pre class="r"><code>meth_df &lt;- read_tsv(&quot;data/methylation_chrX.bed&quot;, 
                    col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;meth&quot;))</code></pre>
<pre><code>Rows: 1105885 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, stop, meth

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>meth_df %&gt;%
  head() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">chrom</th>
<th align="right">start</th>
<th align="right">stop</th>
<th align="right">meth</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chrX</td>
<td align="right">10152</td>
<td align="right">10154</td>
<td align="right">0.576923</td>
</tr>
<tr class="even">
<td align="left">chrX</td>
<td align="right">10232</td>
<td align="right">10234</td>
<td align="right">0.411765</td>
</tr>
<tr class="odd">
<td align="left">chrX</td>
<td align="right">10330</td>
<td align="right">10332</td>
<td align="right">0.750000</td>
</tr>
<tr class="even">
<td align="left">chrX</td>
<td align="right">10334</td>
<td align="right">10336</td>
<td align="right">0.350000</td>
</tr>
<tr class="odd">
<td align="left">chrX</td>
<td align="right">10336</td>
<td align="right">10338</td>
<td align="right">0.300000</td>
</tr>
<tr class="even">
<td align="left">chrX</td>
<td align="right">10364</td>
<td align="right">10366</td>
<td align="right">0.684211</td>
</tr>
</tbody>
</table>
<p>Let’s plot these values by position:</p>
<pre class="r"><code>meth_df %&gt;%
  ggplot(aes(x = start, y = meth)) +
  geom_point() +
  xlab(&quot;Start Position&quot;) +
  ylab(&quot;Measure&quot;)</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Wow, very helpful; let’s take a look at the distribution of the range
of values for the measure:</p>
<pre class="r"><code>meth_df %&gt;%
  ggplot(aes(x = meth)) + 
  geom_boxplot()</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Okay, so if I use the rule &gt;50% = methylated, what does our
distribution of methylated vs unmethylated sites look like?</p>
<pre class="r"><code>meth_df %&gt;%
  mutate(is_meth = meth &gt; 0.5) %&gt;%
  tabyl(is_meth)</code></pre>
<pre><code> is_meth      n   percent
   FALSE 226249 0.2045864
    TRUE 879636 0.7954136</code></pre>
<pre class="r"><code>meth_df[sample(1:nrow(meth_df), 50000, replace = F),] %&gt;%
  mutate(is_meth = as.numeric(meth &gt; 0.5)) %&gt;%
  ggplot(aes(x = start, y = is_meth, colour = factor(is_meth))) +
  geom_point()</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Other things I might want to look at?</p>
<ul>
<li>Runs of methylation</li>
</ul>
</div>
<div id="runs-of-methylation" class="section level2">
<h2>Runs of methylation</h2>
<pre class="r"><code>meth_df &lt;- meth_df %&gt;%
  mutate(is_meth = as.numeric(meth &gt; 0.5))

write_csv(meth_df, file = &quot;output/meth_locations.csv&quot;)</code></pre>
<pre class="r"><code>meth_runs &lt;- function(df){
  final &lt;- data.frame(start = numeric(), end = numeric(), type = numeric(), n = numeric())
  current_start &lt;- df$start[1]
  current_end &lt;- df$stop[1]
  current_type &lt;- df$is_meth[1]
  counts &lt;- 1
  
  for(i in 2:nrow(df)){
    if(df$is_meth[i] == current_type){
      current_end = df$stop[i]
      counts &lt;- counts + 1
    } else{
      final &lt;- bind_rows(final, data.frame(start = current_start, end = current_end, type = current_type, n = counts))
      current_start = df$start[i]
      current_end = df$stop[i]
      current_type = df$is_meth[i]
      counts &lt;- 1
    }
  }
  final &lt;- bind_rows(final, data.frame(start = current_start, end = current_end, type = current_type, n = counts))
  return(final)
}</code></pre>
<pre class="r"><code>runny_meth &lt;- meth_runs(meth_df)</code></pre>
<pre class="r"><code>runny_meth %&gt;%
  ggplot(aes(x = n, y = type, group_by = factor(type), colour = factor(type))) + 
  geom_boxplot()</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>runny_meth %&gt;%
  ggplot(aes(x = start, y = n, colour = factor(type))) +
  geom_point()</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-to-ovary" class="section level2">
<h2>Compare to Ovary</h2>
<pre class="r"><code>ovary_df &lt;- read_tsv(&quot;data/ovary/ovary_chrX.bed&quot;, 
                    col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;meth&quot;))</code></pre>
<pre><code>Rows: 1160115 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, stop, meth

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>all_meth &lt;- full_join(ovary_df, meth_df, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;)) %&gt;%
  rename(ovary = meth.x, testis = meth.y) 

all_meth %&gt;% summary()</code></pre>
<pre><code>    chrom               start                stop               ovary      
 Length:1173205     Min.   :    10152   Min.   :    10154   Min.   :0.000  
 Class :character   1st Qu.: 31437449   1st Qu.: 31437451   1st Qu.:0.564  
 Mode  :character   Median : 71664463   Median : 71664465   Median :0.767  
                    Mean   : 74689884   Mean   : 74689886   Mean   :0.700  
                    3rd Qu.:119024089   3rd Qu.:119024091   3rd Qu.:0.891  
                    Max.   :156029919   Max.   :156029921   Max.   :1.000  
                                                            NA&#39;s   :12989  
     testis         is_meth     
 Min.   :0.00    Min.   :0.0    
 1st Qu.:0.67    1st Qu.:1.0    
 Median :0.92    Median :1.0    
 Mean   :0.75    Mean   :0.8    
 3rd Qu.:1.00    3rd Qu.:1.0    
 Max.   :1.00    Max.   :1.0    
 NA&#39;s   :67180   NA&#39;s   :67180  </code></pre>
<pre class="r"><code>ovary_df %&gt;%
  ggplot(aes(x = meth)) +
  geom_boxplot() +
  ggtitle(&quot;Methylation: Ovary (X Chromosome)&quot;)</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>all_meth %&gt;%
  remove_missing() %&gt;%
  ggplot(aes(x = testis, y = ovary)) + geom_point()</code></pre>
<pre class="r"><code>all_meth %&gt;%
  select(testis, ovary) %&gt;%
  pivot_longer(testis:ovary, names_to = &quot;sample&quot;, values_to = &quot;methylation&quot;) %&gt;%
  ggplot(aes(x = methylation, y= sample, colour = sample)) + geom_boxplot()</code></pre>
<pre><code>Warning: Removed 80169 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>all_meth %&gt;%
  drop_na(ovary, testis) %&gt;%
  mutate(ov_m = ovary &gt; 0.5, ov_t = testis &gt; 0.5) %&gt;%
  select(ov_m, ov_t) %&gt;%
  xtabs(~ ov_m + ov_t, data = .)</code></pre>
<pre><code>       ov_t
ov_m     FALSE   TRUE
  FALSE  96329 125445
  TRUE  122497 748765</code></pre>
</div>
<div id="methylated-annotated-switches" class="section level2">
<h2>Methylated Annotated Switches</h2>
<pre class="r"><code>get_flip_pos &lt;- function(df){
  flip_list &lt;- c()
  in_progress &lt;- FALSE
  for(i in 1:(length(df$pos_start)-1)){
    if(df$pos_end[i] == df$pos_start[i+1]){
      if(!in_progress){
        in_progress &lt;- TRUE
        flip_list &lt;- c(flip_list, df$pos_end[i])
      } else{
        in_progress &lt;- FALSE
      }
    } else{
      in_progress &lt;- FALSE
    }
  }
  return(flip_list)
}</code></pre>
<p>Let’s first take a look at a single pseudo-diploid and see what
things look like.</p>
<pre class="r"><code>eagle_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/eagle/annotated&quot;
shapeit_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/19april22_switch_errors/switch_errors/shapeit/annotated&quot;

id &lt;- 1

s_file &lt;- paste0(shapeit_dir, &quot;/switch_meth_&quot;, id, &quot;.csv&quot;)
e_file &lt;- paste0(eagle_dir, &quot;/switch_meth_&quot;, id, &quot;.csv&quot;)

s_df &lt;- read_csv(s_file)</code></pre>
<pre><code>Rows: 961 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): motif_start, end_motif
dbl (5): pos_start, pos_end, cpg_start, cpg_end, meth

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>e_df &lt;- read_csv(e_file)</code></pre>
<pre><code>Rows: 965 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): motif_start, end_motif
dbl (5): pos_start, pos_end, cpg_start, cpg_end, meth

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>e_flip_list &lt;- get_flip_pos(e_df)
s_flip_list &lt;- get_flip_pos(s_df)

s_df$is_flip &lt;- as.numeric(s_df$pos_start %in% s_flip_list)
e_df$is_flip &lt;- as.numeric(e_df$pos_start %in% e_flip_list)</code></pre>
<pre class="r"><code>s_df %&gt;%
  filter(cpg_start == 1) %&gt;%
  mutate(is_meth = meth &gt; 0.5) %&gt;%
  ggplot(aes(x = meth)) + geom_boxplot()</code></pre>
<p><img src="figure/methylation_explore.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>e_df %&gt;%
  filter(cpg_start == 1) %&gt;%
  mutate(is_meth = meth &gt; 0.5) %&gt;%
  pull(is_meth) %&gt;% table()</code></pre>
<pre><code>.
FALSE  TRUE 
   72     7 </code></pre>
<pre class="r"><code>s_df %&gt;%
  filter(cpg_start == 1) %&gt;%
  filter(is_flip == 1) %&gt;%
  mutate(is_meth = meth &gt; 0.5) %&gt;%
  pull(is_meth) %&gt;% table()</code></pre>
<pre><code>.
FALSE  TRUE 
   35     5 </code></pre>
<pre class="r"><code>e_df %&gt;%
  filter(cpg_start == 1) %&gt;%
  filter(is_flip == 1)%&gt;%
  mutate(is_meth = meth &gt; 0.5) %&gt;%
  pull(is_meth) %&gt;% table()</code></pre>
<pre><code>.
FALSE  TRUE 
   33     3 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] janitor_2.1.0   forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9    
 [5] purrr_0.3.4     readr_2.1.2     tidyr_1.2.0     tibble_3.1.7   
 [9] ggplot2_3.3.6   tidyverse_1.3.1 workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     lubridate_1.8.0  getPass_0.2-2    ps_1.7.0        
 [5] assertthat_0.2.1 rprojroot_2.0.3  digest_0.6.29    utf8_1.2.2      
 [9] cellranger_1.1.0 R6_2.5.1         backports_1.4.1  reprex_2.0.1    
[13] evaluate_0.15    httr_1.4.3       highr_0.9        pillar_1.7.0    
[17] rlang_1.0.2      readxl_1.4.0     rstudioapi_0.13  whisker_0.4     
[21] callr_3.7.0      jquerylib_0.1.4  rmarkdown_2.14   labeling_0.4.2  
[25] bit_4.0.4        munsell_0.5.0    broom_0.8.0      compiler_4.2.0  
[29] httpuv_1.6.5     modelr_0.1.8     xfun_0.31        pkgconfig_2.0.3 
[33] htmltools_0.5.2  tidyselect_1.1.2 fansi_1.0.3      withr_2.5.0     
[37] crayon_1.5.1     tzdb_0.3.0       dbplyr_2.1.1     later_1.3.0     
[41] grid_4.2.0       jsonlite_1.8.0   gtable_0.3.0     lifecycle_1.0.1 
[45] DBI_1.1.2        git2r_0.30.1     magrittr_2.0.3   scales_1.2.0    
[49] vroom_1.5.7      cli_3.3.0        stringi_1.7.6    farver_2.1.0    
[53] fs_1.5.2         promises_1.2.0.1 snakecase_0.11.0 xml2_1.3.3      
[57] bslib_0.3.1      ellipsis_0.3.2   generics_0.1.2   vctrs_0.4.1     
[61] tools_4.2.0      bit64_4.0.5      glue_1.6.2       hms_1.1.1       
[65] parallel_4.2.0   processx_3.5.3   fastmap_1.1.0    yaml_2.3.5      
[69] colorspace_2.0-3 rvest_1.0.2      knitr_1.39       haven_2.5.0     
[73] sass_0.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
