<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2024-02-28" />

<title>results_paper</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/theandyb/phasing">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">results_paper</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2024-02-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>phasing/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220221code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220221)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220221code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220221)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/eagle/annotated/
</td>
<td style="text-align:left;">
output/final_switch_errors/switch_errors/eagle/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/shapeit/annotated/
</td>
<td style="text-align:left;">
output/final_switch_errors/switch_errors/shapeit/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/beagle/annotated/
</td>
<td style="text-align:left;">
output/final_switch_errors/switch_errors/beagle/annotated
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/vcf_n_sites/
</td>
<td style="text-align:left;">
output/final_switch_errors/vcf_n_sites
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/whatshap/
</td>
<td style="text-align:left;">
output/final_switch_errors/whatshap
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/
</td>
<td style="text-align:left;">
output/final_switch_errors/het_loc
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/background_rates/dimer.csv
</td>
<td style="text-align:left;">
output/background_rates/dimer.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/background_rates/3mer.csv
</td>
<td style="text-align:left;">
output/background_rates/3mer.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/phased_size.txt
</td>
<td style="text-align:left;">
output/final_switch_errors/het_loc/phased_size.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingtreecfc53003e3a2f287d5b5163b713d2afcb84c2d01targetblankcfc5300a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/phasing/tree/cfc53003e3a2f287d5b5163b713d2afcb84c2d01" target="_blank">cfc5300</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingtreecfc53003e3a2f287d5b5163b713d2afcb84c2d01targetblankcfc5300a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/phasing/tree/cfc53003e3a2f287d5b5163b713d2afcb84c2d01" target="_blank">cfc5300</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/1kgp/
    Ignored:    data/consensus/
    Ignored:    data/gnomAD/
    Ignored:    data/mask/
    Ignored:    data/ovary/
    Ignored:    data/recomb/
    Ignored:    data/ref/
    Ignored:    data/shapeit/
    Ignored:    output/2023_switch_errors/
    Ignored:    output/X_hets/
    Ignored:    output/admix_switch_errors/
    Ignored:    output/afr_eur_ref/
    Ignored:    output/background_rates/
    Ignored:    output/figures/
    Ignored:    output/filter_switch_errors/
    Ignored:    output/final_switch_errors/
    Ignored:    output/gnomad/
    Ignored:    output/male_only_reference/
    Ignored:    output/nygc/
    Ignored:    output/singleton_switch_errors/
    Ignored:    plink.log
    Ignored:    sandbox/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   code/batch_annotate_het.sh
    Modified:   code/batch_annotate_switch.sh
    Modified:   code/batch_diploid_v3.sh
    Modified:   code/batch_vote_phase.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/results_paper.Rmd</code>) and
HTML (<code>docs/results_paper.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing/blob/cfc53003e3a2f287d5b5163b713d2afcb84c2d01/analysis/results_paper.Rmd" target="_blank">cfc5300</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-02-28
</td>
<td>
wflow_publish("analysis/results_paper.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="loading-results" class="section level2">
<h2>Loading Results</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(janitor)</code></pre>
<pre><code>
Attaching package: &#39;janitor&#39;

The following objects are masked from &#39;package:stats&#39;:

    chisq.test, fisher.test</code></pre>
<pre class="r"><code>source(&quot;code/common_functions.R&quot;)

library(ggsci)
library(reactable)
theme_set(theme_bw(base_size = 18))

eagle_switch_dir &lt;-   &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/eagle/annotated/&quot;
shapeit_switch_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/shapeit/annotated/&quot;
beagle_switch_dir &lt;-  &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/beagle/annotated/&quot;
num_sites_dir &lt;-      &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/vcf_n_sites/&quot;
whatshap_dir &lt;-       &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/whatshap/&quot;
whatshap_dir2 &lt;-      &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/whatshap/&quot;
het_loc_dir &lt;-        &quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/&quot;

pair_info_df &lt;- read_delim(&quot;data/sample_pairs_9aug2023.csv&quot;, col_names = c(&quot;POP&quot;, &quot;ID1&quot;, &quot;ID2&quot;))</code></pre>
<pre><code>Rows: 700 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): POP, ID1, ID2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pair_info_df$SP &lt;- c(rep(&quot;EUR&quot;, 200), rep(&quot;AFR&quot;, 200), rep(&quot;AMR&quot;, 100), rep(&quot;EAS&quot;, 100), rep(&quot;SAS&quot;, 100))

gc_content_1kb &lt;- read_tsv(&quot;data/ref/gc1kb_X_only.bed&quot;)</code></pre>
<pre><code>Rows: 156041 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (1): #1_usercol
dbl (11): 2_usercol, 3_usercol, 4_pct_at, 5_pct_gc, 6_num_A, 7_num_C, 8_num_...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>colnames(gc_content_1kb) &lt;- c(&quot;CHR&quot;, &quot;START&quot;, &quot;END&quot;, &quot;AT&quot;, &quot;GC&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;TOTAL&quot;, &quot;OTHER&quot;, &quot;LENGTH&quot;)
gc_content_1kb  &lt;- gc_content_1kb %&gt;%
  mutate(bin_id = (START / 1000) + 1)

background_dimer &lt;- read_csv(&quot;/net/snowwhite/home/beckandy/research/phasing/output/background_rates/dimer.csv&quot;)</code></pre>
<pre><code>Rows: 16 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): Nucs
dbl (1): N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>background_3mer &lt;- read_csv(&quot;/net/snowwhite/home/beckandy/research/phasing/output/background_rates/3mer.csv&quot;)</code></pre>
<pre><code>Rows: 64 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): Nucs
dbl (1): N

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)

background_cpg_het &lt;- 229868 / 1567042

df_vcftools &lt;- lapply(c(1:700), 
                    function(x){
                      switch_summary(x, eagle_switch_dir, beagle_switch_dir, shapeit_switch_dir, gc_content_1kb, het_loc_dir)
                      }) %&gt;%
  bind_rows()

df_vcftools$pop &lt;- c(rep(&quot;EUR&quot;, 200), rep(&quot;AFR&quot;, 200), rep(&quot;AMR&quot;, 100), rep(&quot;EAS&quot;, 100), rep(&quot;SAS&quot;, 100))

df_wh_eagle &lt;- get_all_whatshap(&quot;eagle&quot;, n = 700, pop = c(rep(&quot;EUR&quot;, 200), rep(&quot;AFR&quot;, 200), rep(&quot;AMR&quot;, 100), rep(&quot;EAS&quot;, 100), rep(&quot;SAS&quot;, 100)))
df_wh_beagle &lt;- get_all_whatshap(&quot;beagle&quot;, n = 700, pop = c(rep(&quot;EUR&quot;, 200), rep(&quot;AFR&quot;, 200), rep(&quot;AMR&quot;, 100), rep(&quot;EAS&quot;, 100), rep(&quot;SAS&quot;, 100)))
df_wh_si &lt;- get_all_whatshap(&quot;shapeit&quot;, n = 700, pop = c(rep(&quot;EUR&quot;, 200), rep(&quot;AFR&quot;, 200), rep(&quot;AMR&quot;, 100), rep(&quot;EAS&quot;, 100), rep(&quot;SAS&quot;, 100)))

par_size &lt;- (2781479 - 10001) + (156030895 - 155701383)
pd_size &lt;- read_tsv(&quot;/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/phased_size.txt&quot;,
                    col_names = c(&quot;id&quot;, &quot;size&quot;), show_col_types = FALSE) %&gt;%
  mutate(size_mb = (size - par_size) / 1e6)

df_wh_eagle$size_mb &lt;- pd_size$size_mb
df_wh_beagle$size_mb &lt;- pd_size$size_mb
df_wh_si$size_mb &lt;- pd_size$size_mb
df_vcftools$size_mb &lt;- pd_size$size_mb</code></pre>
</div>
<div id="short-paragraph-to-set-the-stage-for-what-we-did"
class="section level1">
<h1>Short paragraph to set the stage for what we did</h1>
<p>With our simulated synthetic diploids, we evaluate how well each of
the three methods were able to reconstruct the known haplotypes. Both
switches and flips are tallied across synthetic diploids to identify
which methods introduced the most and least errors of each kind.
Furthermore, we evaluate the overlap of errors between methods by
identifying shared flips and switches within each synthetic diploid.
Finally, we describe the genomic context in which errors occur by
contrasting the density of errors across the chromosome with genomic
features such as GC content and recombination rate.</p>
</div>
<div id="distribution-of-total-errors-across-methods"
class="section level1">
<h1>Distribution of total errors across methods</h1>
<pre class="r"><code>df_vcftools &lt;- df_vcftools %&gt;%
  mutate(n_total_beagle = n_other_beagle + n_flip_beagle,
         n_total_eagle = n_other_eagle + n_flip_eagle,
         n_total_shapeit = n_other_shapeit + n_flip_shapeit) 

df_vcftools %&gt;%
  select(n_total_beagle, n_total_eagle, n_total_shapeit) %&gt;%
  psych::describe()</code></pre>
<pre><code>                vars   n   mean    sd median trimmed   mad min max range  skew
n_total_beagle     1 700 224.92 69.74  228.5  226.29 61.53  31 536   505 -0.04
n_total_eagle      2 700 256.65 72.83  252.0  255.52 65.23  70 635   565  0.45
n_total_shapeit    3 700 216.73 67.48  217.0  217.39 62.27  42 444   402 -0.04
                kurtosis   se
n_total_beagle      0.34 2.64
n_total_eagle       1.41 2.75
n_total_shapeit    -0.12 2.55</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_total_shapeit &lt; .$n_total_eagle)}</code></pre>
<pre><code>[1] 672</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_total_shapeit &lt; .$n_total_beagle)}</code></pre>
<pre><code>[1] 454</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_total_beagle &lt; .$n_total_eagle)}</code></pre>
<pre><code>[1] 645</code></pre>
<p>Averaging across all 700 synthetic diploids, we find that SHAPEIT
tends to introduce fewer total errors overall (mean 216.7, sd 69.74)
than both Eagle (mean 256.65, sd 72.83) and Beagle (224.92, 69.74).
While SHAPEIT does not always introduce fewer total errors in each
synthetic diploid, for 672/700 synthetic diploids it introduces fewer
total errors than Eagle. Beagle also tends to introduce fewer total
errors than Eagle, with 645/700 synthetic diploids having fewer errors
in Beagle than in Eagle. The distributions of total errors is more
comparable between Beagle and SHAPEIT, where SHAPEIT introduces fewer
errors in just over half (454/700) of the synthetic diploids. Within
each synthetic diploid, Eagle introduced on average 31.73 (25.09) more
total errors than beagle and 39.92 (28.28) more errors than SHAPEIT,
while Beagle on average introduced 8.19 (18.80) more total errors than
SHAPEIT.</p>
<pre class="r"><code>df_tmp &lt;- df_vcftools %&gt;%
  mutate(diff_e_b = n_total_eagle - n_total_beagle,
         diff_e_s = n_total_eagle - n_total_shapeit,
         diff_b_s = n_total_beagle - n_total_shapeit)
df_tmp %&gt;%
  select(starts_with(&quot;diff_&quot;)) %&gt;%
  psych::describe()</code></pre>
<pre><code>         vars   n  mean    sd median trimmed   mad min max range skew kurtosis
diff_e_b    1 700 31.73 25.09     28   29.88 22.24 -42 186   228 1.05     2.94
diff_e_s    2 700 39.92 28.28     37   37.27 22.24 -19 211   230 1.54     4.93
diff_b_s    3 700  8.19 18.80      7    7.42 17.79 -52  92   144 0.37     0.66
           se
diff_e_b 0.95
diff_e_s 1.07
diff_b_s 0.71</code></pre>
<pre class="r"><code>t.test(df_tmp$diff_b_s)</code></pre>
<pre><code>
    One Sample t-test

data:  df_tmp$diff_b_s
t = 11.524, df = 699, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 6.792289 9.581996
sample estimates:
mean of x 
 8.187143 </code></pre>
<pre class="r"><code>t.test(df_tmp$diff_e_b)</code></pre>
<pre><code>
    One Sample t-test

data:  df_tmp$diff_e_b
t = 33.46, df = 699, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 29.87084 33.59487
sample estimates:
mean of x 
 31.73286 </code></pre>
<pre class="r"><code>t.test(df_tmp$diff_e_s)</code></pre>
<pre><code>
    One Sample t-test

data:  df_tmp$diff_e_s
t = 37.354, df = 699, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 37.82175 42.01825
sample estimates:
mean of x 
    39.92 </code></pre>
<pre class="r"><code>df_tmp %&gt;%
  ggplot(aes(x = n_total_beagle, y = n_total_eagle, color=pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  theme_classic() + 
  ggtitle(&quot;Total Errors&quot;) +
  xlab(&quot;Beagle&quot;) +
  ylab(&quot;Eagle&quot;)</code></pre>
<p><img src="figure/results_paper.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distributions-of-switch-errors" class="section level1">
<h1>Distributions of Switch Errors</h1>
<pre class="r"><code>df_vcftools %&gt;%
  select(starts_with(&quot;n_other&quot;)) %&gt;%
  psych::describe()</code></pre>
<pre><code>                    vars   n   mean    sd median trimmed   mad min max range
n_other_eagle          1 700 155.77 47.50    154  155.11 43.00  33 367   334
n_other_beagle         2 700 138.23 46.38    139  138.51 44.48  20 309   289
n_other_shapeit        3 700 107.91 43.64    104  106.59 44.48  10 235   225
n_other_cpg_eagle      4 700  30.14 10.50     30   29.83 10.38   2  71    69
n_other_cpg_beagle     5 700  25.38  9.65     25   25.15  8.90   1  59    58
n_other_cpg_shapeit    6 700  17.66  8.73     17   17.12  8.90   1  46    45
                     skew kurtosis   se
n_other_eagle        0.26     0.47 1.80
n_other_beagle      -0.01    -0.09 1.75
n_other_shapeit      0.24    -0.41 1.65
n_other_cpg_eagle    0.36     0.40 0.40
n_other_cpg_beagle   0.24    -0.12 0.36
n_other_cpg_shapeit  0.59     0.10 0.33</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_other_shapeit &lt; .$n_other_eagle)}</code></pre>
<pre><code>[1] 699</code></pre>
<pre class="r"><code>df_vcftools[which(df_vcftools$n_other_eagle &lt; df_vcftools$n_other_shapeit),] %&gt;%
  select(starts_with(&quot;n_other&quot;))</code></pre>
<pre><code>    n_other_eagle n_other_beagle n_other_shapeit n_other_cpg_eagle
464           164            166             170                27
    n_other_cpg_beagle n_other_cpg_shapeit
464                 31                  24</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_other_shapeit &lt; .$n_other_beagle)}</code></pre>
<pre><code>[1] 685</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_other_beagle &lt; .$n_other_eagle)}</code></pre>
<pre><code>[1] 604</code></pre>
<pre class="r"><code>df_tmp &lt;- df_vcftools %&gt;%
  mutate(diff_e_b = n_other_eagle - n_other_beagle,
         diff_e_s = n_other_eagle - n_other_shapeit,
         diff_b_s = n_other_beagle - n_other_shapeit)

df_tmp %&gt;%
  select(starts_with(&quot;diff_&quot;)) %&gt;%
  psych::describe()</code></pre>
<pre><code>         vars   n  mean    sd median trimmed   mad min max range skew kurtosis
diff_e_b    1 700 17.54 16.48     16   16.71 16.31 -30  91   121 0.57     0.79
diff_e_s    2 700 47.86 22.19     45   45.92 17.79  -6 176   182 1.58     5.32
diff_b_s    3 700 30.32 18.32     28   29.04 17.79 -22 118   140 0.79     1.34
           se
diff_e_b 0.62
diff_e_s 0.84
diff_b_s 0.69</code></pre>
<p>Considering flips and non-flip switches as distinct categories of
error, we compare the distributions of both kinds of errors across the
three methods. Similar to total errors, we find that on average SHAPEIT
introduces fewer switches (mean 107.91, sd 43.64) than either Eagle
(mean 155.77, sd 47.50) or Beagle (mean 138.23, sd 46.38). Within
individual synthetic diploids, Eagle on average introduces 47.86 (sd
22.19) switches than SHAPEIT and 17.54 (sd 16.48) more switches than
Beagle, while Begle on average introduces 30.32 (sd 18.32) more switches
than SHAPEIT. Only in 1 of the 700 synthetic diploids does SHAPEIT
introduce more switches than EAGLE, while in 685/700 synthetic diploids
fewer switches are generated by SHAPEIT than in Beagle. Beagle also
introduces fewer switches than Eagle in most synthetic diploids, with
only 83 synthetic diploids having fewer switches in Eagle than in
Beagle.</p>
</div>
<div id="distributions-of-flips" class="section level1">
<h1>Distributions of Flips</h1>
<pre class="r"><code>df_vcftools %&gt;%
  select(starts_with(&quot;n_flip&quot;)) %&gt;%
  psych::describe()</code></pre>
<pre><code>                   vars   n   mean    sd median trimmed   mad min max range
n_flip_eagle          1 700 100.87 29.33   98.5   99.48 25.95  32 268   236
n_flip_beagle         2 700  86.69 27.00   86.0   86.56 25.20  11 227   216
n_flip_shapeit        3 700 108.81 31.46  108.0  108.81 28.17  30 264   234
n_flip_cpg_eagle      4 700  15.64  5.86   15.0   15.27  5.93   2  46    44
n_flip_cpg_beagle     5 700  13.92  5.33   14.0   13.81  5.93   0  35    35
n_flip_cpg_shapeit    6 700  17.54  6.23   17.0   17.43  5.93   2  46    44
                   skew kurtosis   se
n_flip_eagle       0.96     3.13 1.11
n_flip_beagle      0.28     1.14 1.02
n_flip_shapeit     0.26     1.16 1.19
n_flip_cpg_eagle   0.87     2.03 0.22
n_flip_cpg_beagle  0.28    -0.02 0.20
n_flip_cpg_shapeit 0.32     0.51 0.24</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_flip_shapeit &gt;= .$n_flip_eagle)}</code></pre>
<pre><code>[1] 516</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_flip_shapeit &gt; .$n_flip_beagle)}</code></pre>
<pre><code>[1] 691</code></pre>
<pre class="r"><code>df_vcftools %&gt;%
  {sum(.$n_flip_beagle &lt; .$n_flip_eagle)}</code></pre>
<pre><code>[1] 582</code></pre>
<pre class="r"><code>df_tmp &lt;- df_vcftools %&gt;%
  mutate(diff_e_b = n_flip_eagle - n_flip_beagle,
         diff_e_s = n_flip_eagle - n_flip_shapeit,
         diff_b_s = n_flip_beagle - n_flip_shapeit)

df_tmp %&gt;%
  select(starts_with(&quot;diff_&quot;)) %&gt;%
  psych::describe()</code></pre>
<pre><code>         vars   n   mean    sd median trimmed   mad min max range  skew
diff_e_b    1 700  14.19 14.84     13   13.36 13.34 -28  95   123  0.72
diff_e_s    2 700  -7.94 15.56     -9   -8.68 14.83 -49  58   107  0.61
diff_b_s    3 700 -22.13 11.42    -21  -21.56 10.38 -72   4    76 -0.55
         kurtosis   se
diff_e_b     1.61 0.56
diff_e_s     1.06 0.59
diff_b_s     0.53 0.43</code></pre>
<p>In contrast to switch errors, we find that SHAPEIT tends to introduce
more flips errors on average (mean 108.81, sd 31.46) than either Eagle
(mean 100.87, sd 29.33) or Beagle (mean 86.69, sd 27.00). Out of the 700
synthetic diploids, SHAPEIT introduces as many or more flips as Eagle in
516 synthetic diploids and as many or more flips as Beagle in 695
synthetic diploids, while in 582 synthetic diploids Eagle introduces
more flips than Beagle. On average within each synthetic diploid,
SHAPEIT introduces 7.94 (15.56) more flips than Eagle and 22.13 (11.42)
more flips than Beagle, while Eagle introduces 14.19 (14.84) more flips
than Beagle.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] reactable_0.4.4 ggsci_3.0.0     janitor_2.2.0   lubridate_1.9.3
 [5] forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4     purrr_1.0.2    
 [9] readr_2.1.5     tidyr_1.3.1     tibble_3.2.1    ggplot2_3.5.0  
[13] tidyverse_2.0.0 workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.42         bslib_0.6.1       htmlwidgets_1.6.4
 [5] psych_2.4.1       processx_3.8.3    lattice_0.22-5    callr_3.7.5      
 [9] tzdb_0.4.0        vctrs_0.6.5       tools_4.3.2       ps_1.7.6         
[13] generics_0.1.3    parallel_4.3.2    fansi_1.0.6       highr_0.10       
[17] pkgconfig_2.0.3   lifecycle_1.0.4   farver_2.1.1      compiler_4.3.2   
[21] git2r_0.33.0      mnormt_2.1.1      munsell_0.5.0     getPass_0.2-4    
[25] snakecase_0.11.1  httpuv_1.6.14     htmltools_0.5.7   sass_0.4.8       
[29] yaml_2.3.8        later_1.3.2       pillar_1.9.0      crayon_1.5.2     
[33] jquerylib_0.1.4   whisker_0.4.1     cachem_1.0.8      nlme_3.1-164     
[37] tidyselect_1.2.0  digest_0.6.34     stringi_1.8.3     labeling_0.4.3   
[41] rprojroot_2.0.4   fastmap_1.1.1     grid_4.3.2        archive_1.1.7    
[45] colorspace_2.1-0  cli_3.6.2         magrittr_2.0.3    utf8_1.2.4       
[49] withr_3.0.0       scales_1.3.0      promises_1.2.1    bit64_4.0.5      
[53] timechange_0.3.0  rmarkdown_2.25    httr_1.4.7        bit_4.0.5        
[57] hms_1.1.3         evaluate_0.23     knitr_1.45        rlang_1.1.3      
[61] Rcpp_1.0.12       glue_1.7.0        rstudioapi_0.15.0 vroom_1.6.5      
[65] jsonlite_1.8.8    R6_2.5.1          fs_1.6.3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
