---
title: "build_a_vcf"
author: "Andy Beck"
date: "2022-02-28"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(vcfR)
```

Using the list of sites in `data/chrX_sites_28feb2022.tsv`, we filtered the original VCF down to 2 subjects (HG00260 and HG00136) and the variant sites which are outside the PAR and are homozygous in both individuals. Now we'd like to "smash" these haplotypes together to make a pseudo diploid.

Let's first load in our filtered VCF:

```{r, eval=FALSE}
vcf <- read.vcfR("data/chrX_test.vcf")
#chrom <- create.chromR(name='test', vcf=vcf)

df <- vcf@gt %>% as.data.frame()
df <- df %>%
  mutate(FAKE001 = str_c( 
    str_sub( !!as.name("HG00260"), 1, 1), 
    "/",  
    str_sub( !!as.name("HG00136"), 1, 1)) ) %>%
  select(FORMAT, FAKE001)

vcf@gt <- as.matrix(df)

write.vcf(vcf, "data/chrX_test_fake.vcf")
```

