---
title: "results_paper"
author: "Andy Beck"
date: "2024-02-28"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Loading Results

```{r}
library(tidyverse)
library(janitor)
source("code/common_functions.R")

library(ggsci)
library(reactable)
theme_set(theme_bw(base_size = 18))

eagle_switch_dir <-   "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/eagle/annotated/"
shapeit_switch_dir <- "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/shapeit/annotated/"
beagle_switch_dir <-  "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/switch_errors/beagle/annotated/"
num_sites_dir <-      "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/vcf_n_sites/"
whatshap_dir <-       "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/whatshap/"
whatshap_dir2 <-      "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/whatshap/"
het_loc_dir <-        "/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/"

pair_info_df <- read_delim("data/sample_pairs_9aug2023.csv", col_names = c("POP", "ID1", "ID2"))
pair_info_df$SP <- c(rep("EUR", 200), rep("AFR", 200), rep("AMR", 100), rep("EAS", 100), rep("SAS", 100))

gc_content_1kb <- read_tsv("data/ref/gc1kb_X_only.bed")
colnames(gc_content_1kb) <- c("CHR", "START", "END", "AT", "GC", "A", "C", "G", "T", "TOTAL", "OTHER", "LENGTH")
gc_content_1kb  <- gc_content_1kb %>%
  mutate(bin_id = (START / 1000) + 1)

background_dimer <- read_csv("/net/snowwhite/home/beckandy/research/phasing/output/background_rates/dimer.csv")
background_3mer <- read_csv("/net/snowwhite/home/beckandy/research/phasing/output/background_rates/3mer.csv")

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

background_cpg_het <- 229868 / 1567042

df_vcftools <- lapply(c(1:700), 
                    function(x){
                      switch_summary(x, eagle_switch_dir, beagle_switch_dir, shapeit_switch_dir, gc_content_1kb, het_loc_dir)
                      }) %>%
  bind_rows()

df_vcftools$pop <- c(rep("EUR", 200), rep("AFR", 200), rep("AMR", 100), rep("EAS", 100), rep("SAS", 100))

df_wh_eagle <- get_all_whatshap("eagle", n = 700, pop = c(rep("EUR", 200), rep("AFR", 200), rep("AMR", 100), rep("EAS", 100), rep("SAS", 100)))
df_wh_beagle <- get_all_whatshap("beagle", n = 700, pop = c(rep("EUR", 200), rep("AFR", 200), rep("AMR", 100), rep("EAS", 100), rep("SAS", 100)))
df_wh_si <- get_all_whatshap("shapeit", n = 700, pop = c(rep("EUR", 200), rep("AFR", 200), rep("AMR", 100), rep("EAS", 100), rep("SAS", 100)))

par_size <- (2781479 - 10001) + (156030895 - 155701383)
pd_size <- read_tsv("/net/snowwhite/home/beckandy/research/phasing/output/final_switch_errors/het_loc/phased_size.txt",
                    col_names = c("id", "size"), show_col_types = FALSE) %>%
  mutate(size_mb = (size - par_size) / 1e6)

df_wh_eagle$size_mb <- pd_size$size_mb
df_wh_beagle$size_mb <- pd_size$size_mb
df_wh_si$size_mb <- pd_size$size_mb
df_vcftools$size_mb <- pd_size$size_mb
```

# Short paragraph to set the stage for what we did

With our simulated synthetic diploids, we evaluate how well each of the three methods were able to reconstruct the known haplotypes. Both switches and flips are tallied across synthetic diploids to identify which methods introduced the most and least errors of each kind. Furthermore, we evaluate the overlap of errors between methods by identifying shared flips and switches within each synthetic diploid. Finally, we describe the genomic context in which errors occur by contrasting the density of errors across the chromosome with genomic features such as GC content and recombination rate.

# Distribution of total errors across methods

```{r}
df_vcftools <- df_vcftools %>%
  mutate(n_total_beagle = n_other_beagle + n_flip_beagle,
         n_total_eagle = n_other_eagle + n_flip_eagle,
         n_total_shapeit = n_other_shapeit + n_flip_shapeit) 

df_vcftools %>%
  select(n_total_beagle, n_total_eagle, n_total_shapeit) %>%
  psych::describe()

df_vcftools %>%
  {sum(.$n_total_shapeit < .$n_total_eagle)}

df_vcftools %>%
  {sum(.$n_total_shapeit < .$n_total_beagle)}

df_vcftools %>%
  {sum(.$n_total_beagle < .$n_total_eagle)}

df_vcftools %>%
  ggplot(aes(x = n_total_eagle, y = n_total_shapeit, color = pop)) +
  geom_point() +
  xlab("Eagle") +
  ylab("SHAPEIT") +
  geom_abline(slope = 1, intercept = 0) +
  theme_classic() +
  ggtitle("Total Errors") +
  scale_color_manual(values=cbPalette)
```


Averaging across all 700 synthetic diploids, we find that SHAPEIT tends to introduce fewer total errors overall (mean 216.7, sd 69.74) than both Eagle (mean 256.65, sd 72.83) and Beagle (224.92, 69.74). While SHAPEIT does not always introduce fewer total errors in each synthetic diploid, for 672/700 synthetic diploids it introduces fewer total errors than Eagle. Beagle also tends to introduce fewer total errors than Eagle, with 645/700 synthetic diploids having fewer errors in Beagle than in Eagle. The distributions of total errors is more comparable between Beagle and SHAPEIT, where SHAPEIT introduces fewer errors in just over half (454/700) of the synthetic diploids. Within each synthetic diploid, Eagle introduced on average 31.73 (25.09) more total errors than beagle and 39.92 (28.28) more errors than SHAPEIT, while Beagle on average introduced 8.19 (18.80) more total errors than SHAPEIT.

```{r}
df_tmp <- df_vcftools %>%
  mutate(diff_e_b = n_total_eagle - n_total_beagle,
         diff_e_s = n_total_eagle - n_total_shapeit,
         diff_b_s = n_total_beagle - n_total_shapeit)
df_tmp %>%
  select(starts_with("diff_")) %>%
  psych::describe()

t.test(df_tmp$diff_b_s)
t.test(df_tmp$diff_e_b)
t.test(df_tmp$diff_e_s)

df_tmp %>%
  ggplot(aes(x = n_total_beagle, y = n_total_eagle, color=pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  theme_classic() + 
  ggtitle("Total Errors") +
  xlab("Beagle") +
  ylab("Eagle")
```

# Distributions of Switch Errors

```{r}
df_vcftools %>%
  select(starts_with("n_other")) %>%
  psych::describe()

df_vcftools %>%
  {sum(.$n_other_shapeit < .$n_other_eagle)}

df_vcftools[which(df_vcftools$n_other_eagle < df_vcftools$n_other_shapeit),] %>%
  select(starts_with("n_other"))

df_vcftools %>%
  {sum(.$n_other_shapeit < .$n_other_beagle)}

df_vcftools %>%
  {sum(.$n_other_beagle < .$n_other_eagle)}

df_tmp <- df_vcftools %>%
  mutate(diff_e_b = n_other_eagle - n_other_beagle,
         diff_e_s = n_other_eagle - n_other_shapeit,
         diff_b_s = n_other_beagle - n_other_shapeit)

df_tmp %>%
  select(starts_with("diff_")) %>%
  psych::describe()
```


Considering flips and non-flip switches as distinct categories of error, we compare the distributions of both kinds of errors across the three methods. Similar to total errors, we find that on average SHAPEIT introduces fewer switches (mean 107.91, sd 43.64) than either Eagle (mean 155.77, sd 47.50) or Beagle (mean 138.23, sd 46.38). Within individual synthetic diploids, Eagle on average introduces 47.86 (sd 22.19) switches than SHAPEIT and 17.54 (sd 16.48) more switches than Beagle, while Begle on average introduces 30.32 (sd 18.32) more switches than SHAPEIT. Only in 1 of the 700 synthetic diploids does SHAPEIT introduce more switches than EAGLE, while in 685/700 synthetic diploids fewer switches are generated by SHAPEIT than in Beagle. Beagle also introduces fewer switches than Eagle in most synthetic diploids, with only 83 synthetic diploids having fewer switches in Eagle than in Beagle. 

# Distributions of Flips

```{r}
df_vcftools %>%
  select(starts_with("n_flip")) %>%
  psych::describe()

df_vcftools %>%
  {sum(.$n_flip_shapeit >= .$n_flip_eagle)}

df_vcftools %>%
  {sum(.$n_flip_shapeit > .$n_flip_beagle)}

df_vcftools %>%
  {sum(.$n_flip_beagle < .$n_flip_eagle)}

df_tmp <- df_vcftools %>%
  mutate(diff_e_b = n_flip_eagle - n_flip_beagle,
         diff_e_s = n_flip_eagle - n_flip_shapeit,
         diff_b_s = n_flip_beagle - n_flip_shapeit)

df_tmp %>%
  select(starts_with("diff_")) %>%
  psych::describe()
```

In contrast to switch errors, we find that SHAPEIT tends to introduce more flips errors on average (mean 108.81, sd 31.46) than either Eagle (mean 100.87, sd 29.33) or Beagle (mean 86.69, sd 27.00). Out of the 700 synthetic diploids, SHAPEIT introduces as many or more flips as Eagle in 516 synthetic diploids and as many or more flips as Beagle in 695 synthetic diploids, while in 582 synthetic diploids Eagle introduces more flips than Beagle. On average within each synthetic diploid, SHAPEIT introduces 7.94 (15.56) more flips than Eagle and 22.13 (11.42) more flips than Beagle, while Eagle introduces 14.19 (14.84) more flips than Beagle.

# Switches and Flips within populations

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_eagle = mean(n_other_eagle),
            mean_switch_beagle = mean(n_other_beagle),
            mean_switch_shapeit = mean(n_other_shapeit),
            mean_flip_eagle = mean(n_flip_eagle),
            mean_flip_beagle = mean(n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_shapeit)) %>%
  knitr::kable()
```
